// @vw-sre/vws-cdk/lib/modules/vws-ingress-V2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VwsIngressV2 = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_ec2_1 = require("aws-cdk-lib/aws-ec2");
const aws_elasticloadbalancingv2_1 = require("aws-cdk-lib/aws-elasticloadbalancingv2");
const aws_route53_1 = require("aws-cdk-lib/aws-route53");
const aws_s3_1 = require("aws-cdk-lib/aws-s3");
const constructs_1 = require("constructs");
const cfn_login_client_1 = require("../vws/cfn-login-client");
/**
 * VwsIngress construct
 *
 * Creates:
 * - log s3 Bucket for LoadBalancer logs
 * - security group for the LoadBalancer
 * - LoadBalancer
 * - LoadBalancer Listener with or without KUMS login protection
 * - Loadbalancer alias in a given domain
 */
class VwsIngressV2 extends constructs_1.Construct {
    constructor(scope, id, props) {
        super(scope, id);
        this.defaultTargetGroupSet = false;
        this.certificates = props.certificates;
        const logBucket = new aws_s3_1.Bucket(this, 'LogBucket', {
            encryption: aws_s3_1.BucketEncryption.S3_MANAGED,
            lifecycleRules: [
                {
                    id: '1',
                    expiration: props.logRetention ? props.logRetention : aws_cdk_lib_1.Duration.days(3),
                    enabled: true,
                },
                {
                    id: '2',
                    abortIncompleteMultipartUploadAfter: aws_cdk_lib_1.Duration.days(3),
                    enabled: true,
                },
            ],
            accessControl: aws_s3_1.BucketAccessControl.LOG_DELIVERY_WRITE,
        });
        this.securityGroup = new aws_ec2_1.SecurityGroup(this, 'SecurityGroup', {
            vpc: props.vpc,
            allowAllOutbound: true,
        });
        this.securityGroup.addIngressRule(aws_ec2_1.Peer.anyIpv4(), aws_ec2_1.Port.tcp(80));
        this.securityGroup.addIngressRule(aws_ec2_1.Peer.anyIpv4(), aws_ec2_1.Port.tcp(443));
        this.loadBalancer = new aws_elasticloadbalancingv2_1.ApplicationLoadBalancer(this, 'LoadBalancer', {
            vpc: props.vpc,
            internetFacing: true,
            securityGroup: this.securityGroup,
            idleTimeout: props.loadBalancerIdleTimeout || aws_cdk_lib_1.Duration.seconds(10),
        });
        this.loadBalancer.logAccessLogs(logBucket, 'access-logs');
        this.loadBalancer.setAttribute('routing.http.drop_invalid_header_fields.enabled', 'true');
        this.loadBalancer.addRedirect({
            sourcePort: 80,
            sourceProtocol: aws_elasticloadbalancingv2_1.ApplicationProtocol.HTTP,
            targetPort: 443,
            targetProtocol: aws_elasticloadbalancingv2_1.ApplicationProtocol.HTTPS,
        });
        this.additionalIdpCallbackUrls = props.additionalIdpCallbackUrls;
        this.domainName = props.domain.name;
        this.logoutUrls = props.logoutUrl;
        if (props.targetGroup) {
            this.setDefaultTargetGroup(props.targetGroup);
        }
        if (props.additionalTargetGroups) {
            for (let additionalTargetGroup of props.additionalTargetGroups) {
                this.addAdditionalTargetGroup(additionalTargetGroup);
            }
        }
        new aws_route53_1.CfnRecordSet(this, 'LoadBalancerAlias', {
            name: `${props.domain.name}.`,
            type: 'A',
            aliasTarget: {
                dnsName: `dualstack.${this.loadBalancer.loadBalancerDnsName}`,
                hostedZoneId: this.loadBalancer.loadBalancerCanonicalHostedZoneId,
            },
            hostedZoneId: props.domain.hostedZone.hostedZoneId,
            setIdentifier: props.recordSetWeight === undefined ? undefined : this.node.addr,
            weight: props.recordSetWeight === undefined ? undefined : props.recordSetWeight,
        });
        new aws_route53_1.CfnRecordSet(this, 'LoadBalancerAaaaAlias', {
            name: `${props.domain.name}.`,
            type: 'AAAA',
            aliasTarget: {
                dnsName: `dualstack.${this.loadBalancer.loadBalancerDnsName}`,
                hostedZoneId: this.loadBalancer.loadBalancerCanonicalHostedZoneId,
            },
            hostedZoneId: props.domain.hostedZone.hostedZoneId,
            setIdentifier: props.recordSetWeight === undefined ? undefined : this.node.addr,
            weight: props.recordSetWeight === undefined ? undefined : props.recordSetWeight,
        });
        this.node.addValidation({
            validate: () => {
                if (!this.defaultTargetGroupSet) {
                    return ['No default target group set'];
                }
                else {
                    return [];
                }
            },
        });
    }
    static getAuthenticateOidc(loginClient, targetGroup, workaround) {
        if (targetGroup.includeLogin) {
            const action = aws_elasticloadbalancingv2_1.ListenerAction.authenticateOidc({
                authenticationRequestExtraParams: {
                    display: 'page',
                    prompt: 'login',
                },
                authorizationEndpoint: loginClient.authorizationEndpoint,
                clientId: aws_cdk_lib_1.SecretValue.secretsManager(loginClient.secretArn, {
                    jsonField: 'client_id',
                }).unsafeUnwrap(),
                clientSecret: aws_cdk_lib_1.SecretValue.secretsManager(loginClient.secretArn, {
                    jsonField: 'client_secret',
                }),
                issuer: loginClient.issuer,
                scope: 'openid profile email',
                sessionCookieName: 'x-session',
                sessionTimeout: workaround ? undefined : aws_cdk_lib_1.Duration.seconds(28800),
                tokenEndpoint: loginClient.tokenEndpoint,
                userInfoEndpoint: loginClient.userInfoEndpoint,
                onUnauthenticatedRequest: aws_elasticloadbalancingv2_1.UnauthenticatedAction.AUTHENTICATE,
                next: aws_elasticloadbalancingv2_1.ListenerAction.forward([targetGroup.targetGroup]),
            });
            // https://github.com/aws/aws-cdk/issues/12843
            if (workaround) {
                // @ts-ignore
                action.actionJson.authenticateOidcConfig.sessionTimeout = 28800;
            }
            return action;
        }
        else {
            return aws_elasticloadbalancingv2_1.ListenerAction.forward([targetGroup.targetGroup]);
        }
    }
    setDefaultTargetGroup(props) {
        if (this.defaultTargetGroupSet) {
            throw new Error('Default target group already set');
        }
        this.defaultTargetGroupSet = true;
        if (props.includeLogin && !this.loginClient) {
            this.loginClient = this.createLoginClient();
        }
        this.applicationListener = this.loadBalancer.addListener('HttpsListenerV2', {
            certificates: this.certificates,
            protocol: aws_elasticloadbalancingv2_1.ApplicationProtocol.HTTPS,
            sslPolicy: aws_elasticloadbalancingv2_1.SslPolicy.FORWARD_SECRECY_TLS12_RES_GCM,
            defaultAction: VwsIngressV2.getAuthenticateOidc(this.loginClient, props, false),
        });
    }
    addAdditionalTargetGroup(additionalTargetGroup) {
        if (!this.defaultTargetGroupSet) {
            throw new Error('Cannot add additional target group without default target group');
        }
        if (additionalTargetGroup.includeLogin && !this.loginClient) {
            this.loginClient = this.createLoginClient();
        }
        const listenerAction = VwsIngressV2.getAuthenticateOidc(this.loginClient, additionalTargetGroup, true);
        new aws_elasticloadbalancingv2_1.ApplicationListenerRule(this, 'Rule' + additionalTargetGroup.priority, {
            action: listenerAction,
            listener: this.applicationListener,
            conditions: additionalTargetGroup.conditions,
            priority: additionalTargetGroup.priority,
        });
    }
    createLoginClient() {
        var _a, _b;
        const callbackUrls = (_b = (_a = this.additionalIdpCallbackUrls) === null || _a === void 0 ? void 0 : _a.map(url => url + '/oauth2/idpresponse')) !== null && _b !== void 0 ? _b : [];
        return new cfn_login_client_1.CfnLoginClient(this, 'PlatformLogin', {
            callbackUrl: [
                ...callbackUrls,
                `https://${this.domainName}/oauth2/idpresponse`,
            ],
            logoutUrl: this.logoutUrls,
        });
    }
}
exports.VwsIngressV2 = VwsIngressV2;
