// @vw-sre/vws-cdk/lib/modules/vws-ingress.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VwsIngress = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_ec2_1 = require("aws-cdk-lib/aws-ec2");
const aws_elasticloadbalancingv2_1 = require("aws-cdk-lib/aws-elasticloadbalancingv2");
const aws_route53_1 = require("aws-cdk-lib/aws-route53");
const aws_s3_1 = require("aws-cdk-lib/aws-s3");
const constructs_1 = require("constructs");
const cfn_login_client_1 = require("../vws/cfn-login-client");
/**
 * VwsIngress construct
 *
 * Creates:
 * - log s3 Bucket for LoadBalancer logs
 * - security group for the LoadBalancer
 * - LoadBalancer
 * - LoadBalancer Listener with or without KUMS login protection
 * - Loadbalancer alias in a given domain
 *
 * @deprecated
 * @see https://github.com/vw-sre/vws-cdk/tree/main/lib/modules/vws-ingress-V2.ts
 */
class VwsIngress extends constructs_1.Construct {
    constructor(scope, id, props) {
        var _a, _b;
        super(scope, id);
        const logBucket = new aws_s3_1.Bucket(this, 'LogBucket', {
            encryption: aws_s3_1.BucketEncryption.KMS_MANAGED,
            blockPublicAccess: aws_s3_1.BlockPublicAccess.BLOCK_ALL,
            lifecycleRules: [
                {
                    id: '1',
                    expiration: props.logRetention,
                    enabled: true,
                },
                {
                    id: '2',
                    abortIncompleteMultipartUploadAfter: aws_cdk_lib_1.Duration.days(3),
                    enabled: true,
                },
            ],
        });
        this.securityGroup = new aws_ec2_1.SecurityGroup(this, 'SecurityGroup', {
            vpc: props.vpc,
            allowAllOutbound: true,
        });
        this.securityGroup.addIngressRule(aws_ec2_1.Peer.anyIpv4(), aws_ec2_1.Port.tcp(80));
        this.securityGroup.addIngressRule(aws_ec2_1.Peer.anyIpv4(), aws_ec2_1.Port.tcp(443));
        this.loadBalancer = new aws_elasticloadbalancingv2_1.ApplicationLoadBalancer(this, 'LoadBalancer', {
            vpc: props.vpc,
            internetFacing: true,
            securityGroup: this.securityGroup,
            vpcSubnets: props.vpcSubnets,
            idleTimeout: aws_cdk_lib_1.Duration.seconds(10),
        });
        this.loadBalancer.logAccessLogs(logBucket, 'access-logs');
        this.loadBalancer.setAttribute('routing.http.drop_invalid_header_fields.enabled', 'true');
        this.loadBalancer.addRedirect({
            sourcePort: 80,
            sourceProtocol: aws_elasticloadbalancingv2_1.ApplicationProtocol.HTTP,
            targetPort: 443,
            targetProtocol: aws_elasticloadbalancingv2_1.ApplicationProtocol.HTTPS,
        });
        const weight = Math.floor(100 / props.targetGroups.length);
        const weightedTargetGroups = props.targetGroups.map((targetGroup) => ({
            targetGroup,
            weight,
        }));
        const listenerProps = {
            certificates: props.certificates,
            loadBalancer: this.loadBalancer,
            protocol: aws_elasticloadbalancingv2_1.ApplicationProtocol.HTTPS,
            port: 443,
            sslPolicy: aws_elasticloadbalancingv2_1.SslPolicy.FORWARD_SECRECY_TLS12_RES_GCM,
        };
        if (props.includeLogin) {
            const callbackUrls = (_b = (_a = props.additionalIdpCallbackUrls) === null || _a === void 0 ? void 0 : _a.map(url => url + '/oauth2/idpresponse')) !== null && _b !== void 0 ? _b : [];
            const loginClient = new cfn_login_client_1.CfnLoginClient(this, 'PlatformLogin', {
                callbackUrl: [
                    ...callbackUrls,
                    `https://${props.domain.name}/oauth2/idpresponse`,
                ],
            });
            this.applicationListener = new aws_elasticloadbalancingv2_1.ApplicationListener(this, 'HttpsListenerV2', Object.assign(Object.assign({}, listenerProps), { defaultAction: aws_elasticloadbalancingv2_1.ListenerAction.authenticateOidc({
                    authenticationRequestExtraParams: {
                        display: 'page',
                        prompt: 'login',
                    },
                    authorizationEndpoint: loginClient.authorizationEndpoint,
                    clientId: aws_cdk_lib_1.SecretValue.secretsManager(loginClient.secretArn, {
                        jsonField: 'client_id',
                    }).toString(),
                    clientSecret: aws_cdk_lib_1.SecretValue.secretsManager(loginClient.secretArn, {
                        jsonField: 'client_secret',
                    }),
                    issuer: loginClient.issuer,
                    scope: 'openid profile email',
                    sessionCookieName: 'x-session',
                    sessionTimeout: aws_cdk_lib_1.Duration.seconds(28800),
                    tokenEndpoint: loginClient.tokenEndpoint,
                    userInfoEndpoint: loginClient.userInfoEndpoint,
                    onUnauthenticatedRequest: aws_elasticloadbalancingv2_1.UnauthenticatedAction.AUTHENTICATE,
                    next: aws_elasticloadbalancingv2_1.ListenerAction.weightedForward(weightedTargetGroups),
                }) }));
        }
        else {
            this.applicationListener = new aws_elasticloadbalancingv2_1.ApplicationListener(this, 'HttpsListenerV2', Object.assign(Object.assign({}, listenerProps), { defaultAction: aws_elasticloadbalancingv2_1.ListenerAction.weightedForward(weightedTargetGroups) }));
        }
        new aws_route53_1.CfnRecordSet(this, 'LoadBalancerAlias', {
            name: `${props.domain.name}.`,
            type: 'A',
            aliasTarget: {
                dnsName: `dualstack.${this.loadBalancer.loadBalancerDnsName}`,
                hostedZoneId: this.loadBalancer.loadBalancerCanonicalHostedZoneId,
            },
            hostedZoneId: props.domain.hostedZone.hostedZoneId,
        });
        new aws_route53_1.CfnRecordSet(this, 'LoadBalancerAaaaAlias', {
            name: `${props.domain.name}.`,
            type: 'AAAA',
            aliasTarget: {
                dnsName: `dualstack.${this.loadBalancer.loadBalancerDnsName}`,
                hostedZoneId: this.loadBalancer.loadBalancerCanonicalHostedZoneId,
            },
            hostedZoneId: props.domain.hostedZone.hostedZoneId,
        });
    }
}
exports.VwsIngress = VwsIngress;
