// @vw-sre/vws-cdk/lib/modules/vws-slackbot.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VwsSlackbot = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_events_1 = require("aws-cdk-lib/aws-events");
const aws_events_targets_1 = require("aws-cdk-lib/aws-events-targets");
const aws_lambda_1 = require("aws-cdk-lib/aws-lambda");
const aws_sam_1 = require("aws-cdk-lib/aws-sam");
const constructs_1 = require("constructs");
const shared_vpc_1 = require("../vws/shared-vpc");
/**
 * @experimental
 */
class VwsSlackbot extends constructs_1.Construct {
    constructor(scope, id, props) {
        super(scope, id);
        this.scope = scope;
        this.id = id;
        const vpc = new shared_vpc_1.SharedVpc(this, 'SharedVpc').vpc;
        const cfnMapping = new aws_cdk_lib_1.CfnMapping(this, 'Regions', {
            mapping: {
                'us-west-2': {
                    id: 'arn:aws:serverlessrepo:us-west-2:637378239786:applications/vws-aws-slack-bot',
                },
                'us-east-1': {
                    id: 'arn:aws:serverlessrepo:us-east-1:637378239786:applications/vws-aws-slack-bot',
                },
                'eu-west-1': {
                    id: 'arn:aws:serverlessrepo:eu-west-1:637378239786:applications/vws-aws-slack-bot',
                },
                'eu-central-1': {
                    id: 'arn:aws:serverlessrepo:eu-central-1:637378239786:applications/vws-aws-slack-bot',
                },
                'ap-south-1': {
                    id: 'arn:aws:serverlessrepo:ap-south-1:637378239786:applications/vws-aws-slack-bot',
                },
                'ap-southeast-1': {
                    id: 'arn:aws:serverlessrepo:ap-southeast-1:637378239786:applications/vws-aws-slack-bot',
                },
                'ap-northeast-2': {
                    id: 'arn:aws:serverlessrepo:ap-northeast-2:637378239786:applications/vws-aws-slack-bot',
                },
            },
        });
        const cfnApplication = new aws_sam_1.CfnApplication(this, 'SlackBot', {
            location: {
                applicationId: cfnMapping.findInMap(aws_cdk_lib_1.Aws.REGION, 'id'),
                semanticVersion: '1.0.8',
            },
            parameters: {
                VpcId: vpc.vpcId,
                SubnetId: vpc.selectSubnets().subnetIds[0],
                WebhookSecretArn: props.webhookSecret.secretArn,
                ProxyUrl: props.proxyUrl,
            },
        });
        const slackbotLambda = aws_lambda_1.Function.fromFunctionName(this, 'Function', cfnApplication.getAtt('Outputs.LambdaFunction').toString());
        props.conditions.forEach(({ event, message }, index) => {
            new aws_events_1.Rule(this, `Rule${index}`, Object.assign({ enabled: true, targets: [new aws_events_targets_1.LambdaFunction(slackbotLambda, {
                        event: aws_events_1.RuleTargetInput.fromObject({
                            channel: props.slackChannel,
                            username: props.slackUser,
                            attachments: [message],
                        }),
                    })] }, event));
        });
    }
    static createPipelineFailedCondition() {
        return {
            event: {
                description: 'EventRule',
                eventPattern: {
                    detail: {
                        state: ['FAILED'],
                    },
                    detailType: ['CodePipeline Pipeline Execution State Change'],
                    source: ['aws.codepipeline'],
                },
            },
            message: {
                title: `CodePipeline ${aws_events_1.EventField.fromPath('$.detail.state')}`,
                color: 'danger',
                fields: [{
                        title: 'Pipeline',
                        value: aws_events_1.EventField.fromPath('$.detail.pipeline'),
                        short: false,
                    }, {
                        title: 'Account',
                        value: aws_events_1.EventField.account,
                        short: true,
                    }, {
                        title: 'Region',
                        value: aws_events_1.EventField.region,
                        short: true,
                    }],
            },
        };
    }
    static createAlarmCondition(alarmName) {
        return {
            event: {
                description: 'EventRule',
                eventPattern: {
                    detail: {
                        alarmName: [alarmName],
                        state: {
                            value: ['ALARM'],
                        },
                    },
                    detailType: ['CloudWatch Alarm State Change'],
                    source: ['aws.cloudwatch'],
                },
            },
            message: {
                title: `Alarm ${aws_events_1.EventField.fromPath('$.detail.alarmName')}`,
                color: 'danger',
                fields: [{
                        title: 'Reason',
                        value: aws_events_1.EventField.fromPath('$.detail.state.reason'),
                        short: false,
                    }, {
                        title: 'Account',
                        value: aws_events_1.EventField.account,
                        short: true,
                    }, {
                        title: 'Region',
                        value: aws_events_1.EventField.region,
                        short: true,
                    }],
            },
        };
    }
}
exports.VwsSlackbot = VwsSlackbot;
