// @vw-sre/vws-cdk/lib/generated/vws/rvs/vws-rvs-receiverule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnReceiveRule = exports.CfnReceiveRulePropsCharsetConversion = exports.toJson_CfnReceiveRulePropsTransfer = exports.toJson_CfnReceiveRulePropsDestination = exports.toJson_CfnReceiveRuleProps = void 0;
// Generated by cdk-import
const cdk = require("aws-cdk-lib");
/**
 * Converts an object of type 'CfnReceiveRuleProps' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnReceiveRuleProps(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Destination': toJson_CfnReceiveRulePropsDestination(obj.destination),
        'Transfer': toJson_CfnReceiveRulePropsTransfer(obj.transfer),
        'CharsetConversion': obj.charsetConversion,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnReceiveRuleProps = toJson_CfnReceiveRuleProps;
/**
 * Converts an object of type 'CfnReceiveRulePropsDestination' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnReceiveRulePropsDestination(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'S3BucketName': obj.s3BucketName,
        'S3BucketPrefix': obj.s3BucketPrefix,
        'RoleArn': obj.roleArn,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnReceiveRulePropsDestination = toJson_CfnReceiveRulePropsDestination;
/**
 * Converts an object of type 'CfnReceiveRulePropsTransfer' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnReceiveRulePropsTransfer(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'VDSN': obj.vdsn,
        'VDSNPrefix': obj.vdsnPrefix,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnReceiveRulePropsTransfer = toJson_CfnReceiveRulePropsTransfer;
/* eslint-enable max-len, quote-props */
/**
 * Charset conversion that should be applied on the received transfer.
 *
 * @schema CfnReceiveRulePropsCharsetConversion
 */
var CfnReceiveRulePropsCharsetConversion;
(function (CfnReceiveRulePropsCharsetConversion) {
    /** ANSI-IBM037 */
    CfnReceiveRulePropsCharsetConversion["ANSI_IBM037"] = "ANSI-IBM037";
    /** ANSI-IBM273 */
    CfnReceiveRulePropsCharsetConversion["ANSI_IBM273"] = "ANSI-IBM273";
    /** ASCII-IBM037 */
    CfnReceiveRulePropsCharsetConversion["ASCII_IBM037"] = "ASCII-IBM037";
    /** ASCII-IBM273 */
    CfnReceiveRulePropsCharsetConversion["ASCII_IBM273"] = "ASCII-IBM273";
    /** ASCII-IBM500 */
    CfnReceiveRulePropsCharsetConversion["ASCII_IBM500"] = "ASCII-IBM500";
    /** IBM037-ANSI */
    CfnReceiveRulePropsCharsetConversion["IBM037_ANSI"] = "IBM037-ANSI";
    /** IBM037-ASCII */
    CfnReceiveRulePropsCharsetConversion["IBM037_ASCII"] = "IBM037-ASCII";
    /** IBM273-ANSI */
    CfnReceiveRulePropsCharsetConversion["IBM273_ANSI"] = "IBM273-ANSI";
    /** IBM273-ASCII */
    CfnReceiveRulePropsCharsetConversion["IBM273_ASCII"] = "IBM273-ASCII";
    /** IBM500-ASCII */
    CfnReceiveRulePropsCharsetConversion["IBM500_ASCII"] = "IBM500-ASCII";
})(CfnReceiveRulePropsCharsetConversion = exports.CfnReceiveRulePropsCharsetConversion || (exports.CfnReceiveRulePropsCharsetConversion = {}));
/**
 * A CloudFormation `VWS::RVS::ReceiveRule`
 *
 * @cloudformationResource VWS::RVS::ReceiveRule
 * @stability external
 */
class CfnReceiveRule extends cdk.CfnResource {
    /**
     * Create a new `VWS::RVS::ReceiveRule`.
     *
     * @param scope - scope in which this resource is defined
     * @param id    - scoped id of the resource
     * @param props - resource properties
     */
    constructor(scope, id, props) {
        super(scope, id, { type: CfnReceiveRule.CFN_RESOURCE_TYPE_NAME, properties: toJson_CfnReceiveRuleProps(props) });
        this.props = props;
        this.attrId = cdk.Token.asString(this.getAtt('Id'));
    }
}
exports.CfnReceiveRule = CfnReceiveRule;
/**
* The CloudFormation resource type name for this resource class.
*/
CfnReceiveRule.CFN_RESOURCE_TYPE_NAME = "VWS::RVS::ReceiveRule";
