// @vw-sre/vws-cdk/lib/generated/vws/rvs/vws-rvs-sendtransfer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnSendTransfer = exports.CfnSendTransferPropsRecordMode = exports.CfnSendTransferPropsRecordFormat = exports.CfnSendTransferPropsEncryption = exports.CfnSendTransferPropsCharsetConversion = exports.toJson_CfnSendTransferPropsTransfer = exports.toJson_CfnSendTransferPropsRecord = exports.toJson_CfnSendTransferPropsSource = exports.toJson_CfnSendTransferProps = void 0;
// Generated by cdk-import
const cdk = require("aws-cdk-lib");
/**
 * Converts an object of type 'CfnSendTransferProps' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnSendTransferProps(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Source': toJson_CfnSendTransferPropsSource(obj.source),
        'Record': toJson_CfnSendTransferPropsRecord(obj.record),
        'Transfer': toJson_CfnSendTransferPropsTransfer(obj.transfer),
        'CharsetConversion': obj.charsetConversion,
        'Encryption': obj.encryption,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnSendTransferProps = toJson_CfnSendTransferProps;
/**
 * Converts an object of type 'CfnSendTransferPropsSource' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnSendTransferPropsSource(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'S3BucketName': obj.s3BucketName,
        'S3ObjectKey': obj.s3ObjectKey,
        'RoleArn': obj.roleArn,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnSendTransferPropsSource = toJson_CfnSendTransferPropsSource;
/**
 * Converts an object of type 'CfnSendTransferPropsRecord' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnSendTransferPropsRecord(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Format': obj.format,
        'Mode': obj.mode,
        'Length': obj.length,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnSendTransferPropsRecord = toJson_CfnSendTransferPropsRecord;
/**
 * Converts an object of type 'CfnSendTransferPropsTransfer' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnSendTransferPropsTransfer(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'VDSN': obj.vdsn,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnSendTransferPropsTransfer = toJson_CfnSendTransferPropsTransfer;
/* eslint-enable max-len, quote-props */
/**
 * Charset conversion that should be applied before sending the transfer.
 *
 * @schema CfnSendTransferPropsCharsetConversion
 */
var CfnSendTransferPropsCharsetConversion;
(function (CfnSendTransferPropsCharsetConversion) {
    /** ANSI-IBM037 */
    CfnSendTransferPropsCharsetConversion["ANSI_IBM037"] = "ANSI-IBM037";
    /** ANSI-IBM273 */
    CfnSendTransferPropsCharsetConversion["ANSI_IBM273"] = "ANSI-IBM273";
    /** ASCII-IBM037 */
    CfnSendTransferPropsCharsetConversion["ASCII_IBM037"] = "ASCII-IBM037";
    /** ASCII-IBM273 */
    CfnSendTransferPropsCharsetConversion["ASCII_IBM273"] = "ASCII-IBM273";
    /** ASCII-IBM500 */
    CfnSendTransferPropsCharsetConversion["ASCII_IBM500"] = "ASCII-IBM500";
    /** IBM037-ANSI */
    CfnSendTransferPropsCharsetConversion["IBM037_ANSI"] = "IBM037-ANSI";
    /** IBM037-ASCII */
    CfnSendTransferPropsCharsetConversion["IBM037_ASCII"] = "IBM037-ASCII";
    /** IBM273-ANSI */
    CfnSendTransferPropsCharsetConversion["IBM273_ANSI"] = "IBM273-ANSI";
    /** IBM273-ASCII */
    CfnSendTransferPropsCharsetConversion["IBM273_ASCII"] = "IBM273-ASCII";
    /** IBM500-ASCII */
    CfnSendTransferPropsCharsetConversion["IBM500_ASCII"] = "IBM500-ASCII";
})(CfnSendTransferPropsCharsetConversion = exports.CfnSendTransferPropsCharsetConversion || (exports.CfnSendTransferPropsCharsetConversion = {}));
/**
 * Enable encrypted transfer to the target RVS station. The encryption is end-to-end in the RVS OFTP2 protocol and requires the receiving station to have OFTP2 and encryption support.
 *
 * @schema CfnSendTransferPropsEncryption
 */
var CfnSendTransferPropsEncryption;
(function (CfnSendTransferPropsEncryption) {
    /** DISABLED */
    CfnSendTransferPropsEncryption["DISABLED"] = "DISABLED";
    /** ENABLED */
    CfnSendTransferPropsEncryption["ENABLED"] = "ENABLED";
})(CfnSendTransferPropsEncryption = exports.CfnSendTransferPropsEncryption || (exports.CfnSendTransferPropsEncryption = {}));
/**
 * Format of the transferred file.
 *
 * @schema CfnSendTransferPropsRecordFormat
 */
var CfnSendTransferPropsRecordFormat;
(function (CfnSendTransferPropsRecordFormat) {
    /** TEXT */
    CfnSendTransferPropsRecordFormat["TEXT"] = "TEXT";
    /** FIXED */
    CfnSendTransferPropsRecordFormat["FIXED"] = "FIXED";
    /** VARIABLE */
    CfnSendTransferPropsRecordFormat["VARIABLE"] = "VARIABLE";
    /** UNSTRUCTURED */
    CfnSendTransferPropsRecordFormat["UNSTRUCTURED"] = "UNSTRUCTURED";
})(CfnSendTransferPropsRecordFormat = exports.CfnSendTransferPropsRecordFormat || (exports.CfnSendTransferPropsRecordFormat = {}));
/**
 * Specifies if the transferred file is a text file or a binary file. Only valid for format FIXED or VARIABLE.
 *
 * @schema CfnSendTransferPropsRecordMode
 */
var CfnSendTransferPropsRecordMode;
(function (CfnSendTransferPropsRecordMode) {
    /** TXT */
    CfnSendTransferPropsRecordMode["TXT"] = "TXT";
    /** BIN */
    CfnSendTransferPropsRecordMode["BIN"] = "BIN";
})(CfnSendTransferPropsRecordMode = exports.CfnSendTransferPropsRecordMode || (exports.CfnSendTransferPropsRecordMode = {}));
/**
 * A CloudFormation `VWS::RVS::SendTransfer`
 *
 * @cloudformationResource VWS::RVS::SendTransfer
 * @stability external
 */
class CfnSendTransfer extends cdk.CfnResource {
    /**
     * Create a new `VWS::RVS::SendTransfer`.
     *
     * @param scope - scope in which this resource is defined
     * @param id    - scoped id of the resource
     * @param props - resource properties
     */
    constructor(scope, id, props) {
        super(scope, id, { type: CfnSendTransfer.CFN_RESOURCE_TYPE_NAME, properties: toJson_CfnSendTransferProps(props) });
        this.props = props;
        this.attrId = cdk.Token.asString(this.getAtt('Id'));
        this.attrEventBusArn = cdk.Token.asString(this.getAtt('EventBusArn'));
    }
}
exports.CfnSendTransfer = CfnSendTransfer;
/**
* The CloudFormation resource type name for this resource class.
*/
CfnSendTransfer.CFN_RESOURCE_TYPE_NAME = "VWS::RVS::SendTransfer";
