// @vw-sre/vws-cdk/lib/generated/vws/vpc/vws-vpc-shared.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnShared = exports.toJson_CfnSharedProps = void 0;
// Generated by cdk-import
const cdk = require("aws-cdk-lib");
/**
 * Converts an object of type 'CfnSharedProps' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnSharedProps(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {};
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnSharedProps = toJson_CfnSharedProps;
/* eslint-enable max-len, quote-props */
/**
 * A CloudFormation `VWS::VPC::Shared`
 *
 * @cloudformationResource VWS::VPC::Shared
 * @stability external
 */
class CfnShared extends cdk.CfnResource {
    /**
     * Create a new `VWS::VPC::Shared`.
     *
     * @param scope - scope in which this resource is defined
     * @param id    - scoped id of the resource
     * @param props - resource properties
     */
    constructor(scope, id, props) {
        super(scope, id, { type: CfnShared.CFN_RESOURCE_TYPE_NAME, properties: toJson_CfnSharedProps(props) });
        this.props = props;
        this.attrId = cdk.Token.asString(this.getAtt('Id'));
        this.attrVpcId = cdk.Token.asString(this.getAtt('VpcId'));
        this.attrSubnets = cdk.Token.asList(this.getAtt('Subnets'));
    }
}
exports.CfnShared = CfnShared;
/**
* The CloudFormation resource type name for this resource class.
*/
CfnShared.CFN_RESOURCE_TYPE_NAME = "VWS::VPC::Shared";
