// @vw-sre/vws-cdk/lib/generated/vws/vpc/vws-vpc-standard-module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toJson_CfnStandardModulePropsResourcesServiceNacl = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService6 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService5 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService4 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService3 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService2 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService1 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp6 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp5 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp4 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp3 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp2 = exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp1 = exports.toJson_CfnStandardModulePropsResourcesRouteTablePrivate = exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation6 = exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation5 = exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation4 = exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation3 = exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation2 = exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation1 = exports.toJson_CfnStandardModulePropsResourcesRouteTablePublic = exports.toJson_CfnStandardModulePropsResourcesRouteIgw = exports.toJson_CfnStandardModulePropsResourcesIgwAttachment = exports.toJson_CfnStandardModulePropsResourcesIgw = exports.toJson_CfnStandardModulePropsResourcesServiceSubnet6 = exports.toJson_CfnStandardModulePropsResourcesPublicSubnet6 = exports.toJson_CfnStandardModulePropsResourcesAppSubnet6 = exports.toJson_CfnStandardModulePropsResourcesServiceSubnet5 = exports.toJson_CfnStandardModulePropsResourcesPublicSubnet5 = exports.toJson_CfnStandardModulePropsResourcesAppSubnet5 = exports.toJson_CfnStandardModulePropsResourcesServiceSubnet4 = exports.toJson_CfnStandardModulePropsResourcesPublicSubnet4 = exports.toJson_CfnStandardModulePropsResourcesAppSubnet4 = exports.toJson_CfnStandardModulePropsResourcesServiceSubnet3 = exports.toJson_CfnStandardModulePropsResourcesPublicSubnet3 = exports.toJson_CfnStandardModulePropsResourcesAppSubnet3 = exports.toJson_CfnStandardModulePropsResourcesServiceSubnet2 = exports.toJson_CfnStandardModulePropsResourcesPublicSubnet2 = exports.toJson_CfnStandardModulePropsResourcesAppSubnet2 = exports.toJson_CfnStandardModulePropsResourcesServiceSubnet1 = exports.toJson_CfnStandardModulePropsResourcesPublicSubnet1 = exports.toJson_CfnStandardModulePropsResourcesAppSubnet1 = exports.toJson_CfnStandardModulePropsResourcesVpc = exports.toJson_CfnStandardModulePropsResourcesServiceSubnets = exports.toJson_CfnStandardModulePropsResourcesAppSubnets = exports.toJson_CfnStandardModulePropsResourcesPublicSubnets = exports.toJson_CfnStandardModulePropsParametersParamCidrBlock = exports.toJson_CfnStandardModulePropsResources = exports.toJson_CfnStandardModulePropsParameters = exports.toJson_CfnStandardModuleProps = void 0;
exports.CfnStandardModule = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress6 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress5 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress4 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress3 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress2 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress1 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress6 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress6 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress5 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress5 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress4 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress4 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress3 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress3 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress2 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress2 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress1 = exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress1 = exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation6 = exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation5 = exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation4 = exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation3 = exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation2 = exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation1 = void 0;
// Generated by cdk-import
const cdk = require("aws-cdk-lib");
/**
 * Converts an object of type 'CfnStandardModuleProps' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModuleProps(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Parameters': toJson_CfnStandardModulePropsParameters(obj.parameters),
        'Resources': toJson_CfnStandardModulePropsResources(obj.resources),
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModuleProps = toJson_CfnStandardModuleProps;
/**
 * Converts an object of type 'CfnStandardModulePropsParameters' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsParameters(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'ParamCidrBlock': toJson_CfnStandardModulePropsParametersParamCidrBlock(obj.paramCidrBlock),
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsParameters = toJson_CfnStandardModulePropsParameters;
/**
 * Converts an object of type 'CfnStandardModulePropsResources' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResources(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'PublicSubnets': toJson_CfnStandardModulePropsResourcesPublicSubnets(obj.publicSubnets),
        'AppSubnets': toJson_CfnStandardModulePropsResourcesAppSubnets(obj.appSubnets),
        'ServiceSubnets': toJson_CfnStandardModulePropsResourcesServiceSubnets(obj.serviceSubnets),
        'Vpc': toJson_CfnStandardModulePropsResourcesVpc(obj.vpc),
        'AppSubnet1': toJson_CfnStandardModulePropsResourcesAppSubnet1(obj.appSubnet1),
        'PublicSubnet1': toJson_CfnStandardModulePropsResourcesPublicSubnet1(obj.publicSubnet1),
        'ServiceSubnet1': toJson_CfnStandardModulePropsResourcesServiceSubnet1(obj.serviceSubnet1),
        'AppSubnet2': toJson_CfnStandardModulePropsResourcesAppSubnet2(obj.appSubnet2),
        'PublicSubnet2': toJson_CfnStandardModulePropsResourcesPublicSubnet2(obj.publicSubnet2),
        'ServiceSubnet2': toJson_CfnStandardModulePropsResourcesServiceSubnet2(obj.serviceSubnet2),
        'AppSubnet3': toJson_CfnStandardModulePropsResourcesAppSubnet3(obj.appSubnet3),
        'PublicSubnet3': toJson_CfnStandardModulePropsResourcesPublicSubnet3(obj.publicSubnet3),
        'ServiceSubnet3': toJson_CfnStandardModulePropsResourcesServiceSubnet3(obj.serviceSubnet3),
        'AppSubnet4': toJson_CfnStandardModulePropsResourcesAppSubnet4(obj.appSubnet4),
        'PublicSubnet4': toJson_CfnStandardModulePropsResourcesPublicSubnet4(obj.publicSubnet4),
        'ServiceSubnet4': toJson_CfnStandardModulePropsResourcesServiceSubnet4(obj.serviceSubnet4),
        'AppSubnet5': toJson_CfnStandardModulePropsResourcesAppSubnet5(obj.appSubnet5),
        'PublicSubnet5': toJson_CfnStandardModulePropsResourcesPublicSubnet5(obj.publicSubnet5),
        'ServiceSubnet5': toJson_CfnStandardModulePropsResourcesServiceSubnet5(obj.serviceSubnet5),
        'AppSubnet6': toJson_CfnStandardModulePropsResourcesAppSubnet6(obj.appSubnet6),
        'PublicSubnet6': toJson_CfnStandardModulePropsResourcesPublicSubnet6(obj.publicSubnet6),
        'ServiceSubnet6': toJson_CfnStandardModulePropsResourcesServiceSubnet6(obj.serviceSubnet6),
        'Igw': toJson_CfnStandardModulePropsResourcesIgw(obj.igw),
        'IgwAttachment': toJson_CfnStandardModulePropsResourcesIgwAttachment(obj.igwAttachment),
        'RouteIgw': toJson_CfnStandardModulePropsResourcesRouteIgw(obj.routeIgw),
        'RouteTablePublic': toJson_CfnStandardModulePropsResourcesRouteTablePublic(obj.routeTablePublic),
        'SubnetRouteTableAssociation1': toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation1(obj.subnetRouteTableAssociation1),
        'SubnetRouteTableAssociation2': toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation2(obj.subnetRouteTableAssociation2),
        'SubnetRouteTableAssociation3': toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation3(obj.subnetRouteTableAssociation3),
        'SubnetRouteTableAssociation4': toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation4(obj.subnetRouteTableAssociation4),
        'SubnetRouteTableAssociation5': toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation5(obj.subnetRouteTableAssociation5),
        'SubnetRouteTableAssociation6': toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation6(obj.subnetRouteTableAssociation6),
        'RouteTablePrivate': toJson_CfnStandardModulePropsResourcesRouteTablePrivate(obj.routeTablePrivate),
        'PrivateRouteTableAssociationApp1': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp1(obj.privateRouteTableAssociationApp1),
        'PrivateRouteTableAssociationApp2': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp2(obj.privateRouteTableAssociationApp2),
        'PrivateRouteTableAssociationApp3': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp3(obj.privateRouteTableAssociationApp3),
        'PrivateRouteTableAssociationApp4': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp4(obj.privateRouteTableAssociationApp4),
        'PrivateRouteTableAssociationApp5': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp5(obj.privateRouteTableAssociationApp5),
        'PrivateRouteTableAssociationApp6': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp6(obj.privateRouteTableAssociationApp6),
        'PrivateRouteTableAssociationService1': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService1(obj.privateRouteTableAssociationService1),
        'PrivateRouteTableAssociationService2': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService2(obj.privateRouteTableAssociationService2),
        'PrivateRouteTableAssociationService3': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService3(obj.privateRouteTableAssociationService3),
        'PrivateRouteTableAssociationService4': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService4(obj.privateRouteTableAssociationService4),
        'PrivateRouteTableAssociationService5': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService5(obj.privateRouteTableAssociationService5),
        'PrivateRouteTableAssociationService6': toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService6(obj.privateRouteTableAssociationService6),
        'ServiceNacl': toJson_CfnStandardModulePropsResourcesServiceNacl(obj.serviceNacl),
        'ServiceNaclAssociation1': toJson_CfnStandardModulePropsResourcesServiceNaclAssociation1(obj.serviceNaclAssociation1),
        'ServiceNaclAssociation2': toJson_CfnStandardModulePropsResourcesServiceNaclAssociation2(obj.serviceNaclAssociation2),
        'ServiceNaclAssociation3': toJson_CfnStandardModulePropsResourcesServiceNaclAssociation3(obj.serviceNaclAssociation3),
        'ServiceNaclAssociation4': toJson_CfnStandardModulePropsResourcesServiceNaclAssociation4(obj.serviceNaclAssociation4),
        'ServiceNaclAssociation5': toJson_CfnStandardModulePropsResourcesServiceNaclAssociation5(obj.serviceNaclAssociation5),
        'ServiceNaclAssociation6': toJson_CfnStandardModulePropsResourcesServiceNaclAssociation6(obj.serviceNaclAssociation6),
        'ServiceAllowAppIngress1': toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress1(obj.serviceAllowAppIngress1),
        'ServiceAllowAppEgress1': toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress1(obj.serviceAllowAppEgress1),
        'ServiceAllowAppIngress2': toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress2(obj.serviceAllowAppIngress2),
        'ServiceAllowAppEgress2': toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress2(obj.serviceAllowAppEgress2),
        'ServiceAllowAppIngress3': toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress3(obj.serviceAllowAppIngress3),
        'ServiceAllowAppEgress3': toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress3(obj.serviceAllowAppEgress3),
        'ServiceAllowAppIngress4': toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress4(obj.serviceAllowAppIngress4),
        'ServiceAllowAppEgress4': toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress4(obj.serviceAllowAppEgress4),
        'ServiceAllowAppIngress5': toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress5(obj.serviceAllowAppIngress5),
        'ServiceAllowAppEgress5': toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress5(obj.serviceAllowAppEgress5),
        'ServiceAllowAppIngress6': toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress6(obj.serviceAllowAppIngress6),
        'ServiceAllowAppEgress6': toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress6(obj.serviceAllowAppEgress6),
        'ServiceAllowAppVPCEIngress1': toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress1(obj.serviceAllowAppVpceIngress1),
        'ServiceAllowAppVPCEIngress2': toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress2(obj.serviceAllowAppVpceIngress2),
        'ServiceAllowAppVPCEIngress3': toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress3(obj.serviceAllowAppVpceIngress3),
        'ServiceAllowAppVPCEIngress4': toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress4(obj.serviceAllowAppVpceIngress4),
        'ServiceAllowAppVPCEIngress5': toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress5(obj.serviceAllowAppVpceIngress5),
        'ServiceAllowAppVPCEIngress6': toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress6(obj.serviceAllowAppVpceIngress6),
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResources = toJson_CfnStandardModulePropsResources;
/**
 * Converts an object of type 'CfnStandardModulePropsParametersParamCidrBlock' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsParametersParamCidrBlock(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsParametersParamCidrBlock = toJson_CfnStandardModulePropsParametersParamCidrBlock;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPublicSubnets' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPublicSubnets(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPublicSubnets = toJson_CfnStandardModulePropsResourcesPublicSubnets;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesAppSubnets' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesAppSubnets(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesAppSubnets = toJson_CfnStandardModulePropsResourcesAppSubnets;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceSubnets' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceSubnets(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceSubnets = toJson_CfnStandardModulePropsResourcesServiceSubnets;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesVpc' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesVpc(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesVpc = toJson_CfnStandardModulePropsResourcesVpc;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesAppSubnet1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesAppSubnet1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesAppSubnet1 = toJson_CfnStandardModulePropsResourcesAppSubnet1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPublicSubnet1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPublicSubnet1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPublicSubnet1 = toJson_CfnStandardModulePropsResourcesPublicSubnet1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceSubnet1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceSubnet1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceSubnet1 = toJson_CfnStandardModulePropsResourcesServiceSubnet1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesAppSubnet2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesAppSubnet2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesAppSubnet2 = toJson_CfnStandardModulePropsResourcesAppSubnet2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPublicSubnet2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPublicSubnet2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPublicSubnet2 = toJson_CfnStandardModulePropsResourcesPublicSubnet2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceSubnet2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceSubnet2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceSubnet2 = toJson_CfnStandardModulePropsResourcesServiceSubnet2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesAppSubnet3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesAppSubnet3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesAppSubnet3 = toJson_CfnStandardModulePropsResourcesAppSubnet3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPublicSubnet3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPublicSubnet3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPublicSubnet3 = toJson_CfnStandardModulePropsResourcesPublicSubnet3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceSubnet3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceSubnet3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceSubnet3 = toJson_CfnStandardModulePropsResourcesServiceSubnet3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesAppSubnet4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesAppSubnet4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesAppSubnet4 = toJson_CfnStandardModulePropsResourcesAppSubnet4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPublicSubnet4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPublicSubnet4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPublicSubnet4 = toJson_CfnStandardModulePropsResourcesPublicSubnet4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceSubnet4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceSubnet4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceSubnet4 = toJson_CfnStandardModulePropsResourcesServiceSubnet4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesAppSubnet5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesAppSubnet5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesAppSubnet5 = toJson_CfnStandardModulePropsResourcesAppSubnet5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPublicSubnet5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPublicSubnet5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPublicSubnet5 = toJson_CfnStandardModulePropsResourcesPublicSubnet5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceSubnet5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceSubnet5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceSubnet5 = toJson_CfnStandardModulePropsResourcesServiceSubnet5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesAppSubnet6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesAppSubnet6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesAppSubnet6 = toJson_CfnStandardModulePropsResourcesAppSubnet6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPublicSubnet6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPublicSubnet6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPublicSubnet6 = toJson_CfnStandardModulePropsResourcesPublicSubnet6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceSubnet6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceSubnet6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceSubnet6 = toJson_CfnStandardModulePropsResourcesServiceSubnet6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesIgw' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesIgw(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesIgw = toJson_CfnStandardModulePropsResourcesIgw;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesIgwAttachment' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesIgwAttachment(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesIgwAttachment = toJson_CfnStandardModulePropsResourcesIgwAttachment;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesRouteIgw' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesRouteIgw(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesRouteIgw = toJson_CfnStandardModulePropsResourcesRouteIgw;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesRouteTablePublic' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesRouteTablePublic(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesRouteTablePublic = toJson_CfnStandardModulePropsResourcesRouteTablePublic;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesSubnetRouteTableAssociation1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation1 = toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesSubnetRouteTableAssociation2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation2 = toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesSubnetRouteTableAssociation3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation3 = toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesSubnetRouteTableAssociation4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation4 = toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesSubnetRouteTableAssociation5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation5 = toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesSubnetRouteTableAssociation6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation6 = toJson_CfnStandardModulePropsResourcesSubnetRouteTableAssociation6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesRouteTablePrivate' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesRouteTablePrivate(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesRouteTablePrivate = toJson_CfnStandardModulePropsResourcesRouteTablePrivate;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp1 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp2 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp3 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp4 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp5 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp6 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationApp6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationService1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService1 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationService2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService2 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationService3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService3 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationService4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService4 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationService5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService5 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesPrivateRouteTableAssociationService6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService6 = toJson_CfnStandardModulePropsResourcesPrivateRouteTableAssociationService6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceNacl' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceNacl(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceNacl = toJson_CfnStandardModulePropsResourcesServiceNacl;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceNaclAssociation1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceNaclAssociation1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation1 = toJson_CfnStandardModulePropsResourcesServiceNaclAssociation1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceNaclAssociation2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceNaclAssociation2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation2 = toJson_CfnStandardModulePropsResourcesServiceNaclAssociation2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceNaclAssociation3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceNaclAssociation3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation3 = toJson_CfnStandardModulePropsResourcesServiceNaclAssociation3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceNaclAssociation4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceNaclAssociation4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation4 = toJson_CfnStandardModulePropsResourcesServiceNaclAssociation4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceNaclAssociation5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceNaclAssociation5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation5 = toJson_CfnStandardModulePropsResourcesServiceNaclAssociation5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceNaclAssociation6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceNaclAssociation6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceNaclAssociation6 = toJson_CfnStandardModulePropsResourcesServiceNaclAssociation6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppIngress1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress1 = toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppEgress1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress1 = toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppIngress2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress2 = toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppEgress2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress2 = toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppIngress3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress3 = toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppEgress3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress3 = toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppIngress4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress4 = toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppEgress4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress4 = toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppIngress5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress5 = toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppEgress5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress5 = toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppIngress6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress6 = toJson_CfnStandardModulePropsResourcesServiceAllowAppIngress6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppEgress6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress6 = toJson_CfnStandardModulePropsResourcesServiceAllowAppEgress6;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppVpceIngress1' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress1(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress1 = toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress1;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppVpceIngress2' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress2(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress2 = toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress2;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppVpceIngress3' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress3(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress3 = toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress3;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppVpceIngress4' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress4(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress4 = toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress4;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppVpceIngress5' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress5(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress5 = toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress5;
/**
 * Converts an object of type 'CfnStandardModulePropsResourcesServiceAllowAppVpceIngress6' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress6(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Type': obj.type,
        'Properties': obj.properties,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress6 = toJson_CfnStandardModulePropsResourcesServiceAllowAppVpceIngress6;
/* eslint-enable max-len, quote-props */
/**
 * A CloudFormation `VWS::VPC::Standard::MODULE`
 *
 * @cloudformationResource VWS::VPC::Standard::MODULE
 * @stability external
 */
class CfnStandardModule extends cdk.CfnResource {
    /**
     * Create a new `VWS::VPC::Standard::MODULE`.
     *
     * @param scope - scope in which this resource is defined
     * @param id    - scoped id of the resource
     * @param props - resource properties
     */
    constructor(scope, id, props) {
        super(scope, id, { type: CfnStandardModule.CFN_RESOURCE_TYPE_NAME, properties: toJson_CfnStandardModuleProps(props) });
        this.props = props;
    }
}
exports.CfnStandardModule = CfnStandardModule;
/**
* The CloudFormation resource type name for this resource class.
*/
CfnStandardModule.CFN_RESOURCE_TYPE_NAME = "VWS::VPC::Standard::MODULE";
