// @vw-sre/vws-cdk/lib/vws/cfn-shared-vpc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnSharedVpc = void 0;
/**
 * Provides access to a shared VWS managed VPC for utility lambdas and other comparable workloads.
 *
 * Provided endpoints
 * - AWS S3
 * - AWS DynamoDB
 * - AWS SecretsManager
 * - AWS CloudWatch Logs
 * - AWS CodeBuild
 * - AWS CodeDepoy
 * - VWS Proxy
 *
 */
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_ec2_1 = require("aws-cdk-lib/aws-ec2");
/**
 * @deprecated
 * @see https://github.com/vw-sre/vws-cdk/tree/main/lib/vws/shared-vpc.ts
 */
class CfnSharedVpc extends aws_cdk_lib_1.CfnResource {
    constructor(scope, id) {
        super(scope, id, {
            type: 'VWS::VPC::Shared',
        });
        this.plainId = id;
    }
    /**
       * returns the sharedVPC id
       */
    get id() {
        return super.getAtt('Id').toString();
    }
    /**
       * return the id ref of the underlying vpc
       */
    get vpcId() {
        return super.getAtt('VpcId').toString();
    }
    /**
       * return a reference to a list of underlying subnets
       */
    get subnetIds() {
        return super.getAtt('Subnets');
    }
    /**
     * returns a VPC object, with isolated subnets.
     * Due to CDK you have to provide the availability zones for which you would like to get references to.
     *
     * Tested with eu-west-1 with ["eu-west-1a", "eu-west-1b", "eu-west-1c"].
     * You can also use CDKs `stack.availabilityZones`, still this would only provide you two azs.
     *
     * @deprecated
     * @see https://github.com/vw-sre/vws-cdk/tree/main/lib/vws/shared-vpc.ts
     */
    vpc(availabilityZones) {
        const subnetIdList = aws_cdk_lib_1.Token.asList(this.subnetIds);
        let subnetIds = [];
        availabilityZones.forEach((_, index) => {
            subnetIds.push(aws_cdk_lib_1.Fn.select(index, subnetIdList));
        });
        return aws_ec2_1.Vpc.fromVpcAttributes(this, `${this.plainId}-Vpc`, {
            vpcId: this.vpcId,
            availabilityZones: availabilityZones,
            isolatedSubnetIds: subnetIds,
        });
    }
}
exports.CfnSharedVpc = CfnSharedVpc;
