// @vw-sre/vws-cdk/lib/vws/cfn-domain.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnDomain = void 0;
/**
 * @field Name: Name of the subdomain
 * @field Root: root domain to allocate the name with
 * @field Authorization: Serenity route migration authorization
 */
const aws_cdk_lib_1 = require("aws-cdk-lib");
/**
 * Domain Allocates a vwapps.run subdomain
 *
 * @deprecated
 * @see https://github.com/vw-sre/vws-cdk/tree/main/lib/generated/vws/domain/vws-domain-name.ts
 */
class CfnDomain extends aws_cdk_lib_1.CfnResource {
    constructor(scope, id, props) {
        super(scope, id, {
            type: 'VWS::Domain::Name',
            properties: {
                Domain: {
                    Random: aws_cdk_lib_1.Lazy.any({ produce: () => !(props === null || props === void 0 ? void 0 : props.name) ? true : undefined }),
                    Name: aws_cdk_lib_1.Lazy.string({ produce: () => props === null || props === void 0 ? void 0 : props.name }),
                    Root: aws_cdk_lib_1.Lazy.string({ produce: () => props === null || props === void 0 ? void 0 : props.root }),
                    Authorization: aws_cdk_lib_1.Lazy.string({ produce: () => props === null || props === void 0 ? void 0 : props.authorization }),
                },
            },
        });
    }
    /**
       * name returns the FQDN of the domain
       */
    get name() {
        return super.ref;
    }
}
exports.CfnDomain = CfnDomain;
