// @vw-sre/vws-cdk/lib/vws/cfn-login-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnLoginClient = void 0;
/**
 * @field CallbackUrl: list of idp callback urls
 * @field LogoutUrl: list of allowed logout redirect urls
 */
const aws_cdk_lib_1 = require("aws-cdk-lib");
/**
 * IDPLogin enables you to protect your application automatically with a K-UMS login.
 *
 * @deprecated
 * @see https://github.com/vw-sre/vws-cdk/tree/main/lib/generated/vws/cloudfoundry/login/vws-login-client.ts
 */
class CfnLoginClient extends aws_cdk_lib_1.CfnResource {
    constructor(scope, id, properties) {
        super(scope, id, {
            type: 'VWS::Login::Client',
            properties: {
                CallbackUrl: properties.callbackUrl,
                LogoutUrl: properties.logoutUrl,
            },
        });
    }
    /**
       * OIDC client credentials SecretsManager secret arn
       */
    get secretArn() {
        return super.getAtt('SecretArn').toString();
    }
    /**
       * authorization endpoint
       */
    get issuer() {
        return super.getAtt('Issuer').toString();
    }
    /**
       * authorization endpoint
       */
    get authorizationEndpoint() {
        return super.getAtt('AuthorizationEndpoint').toString();
    }
    /**
       * token endpoint
       */
    get tokenEndpoint() {
        return super.getAtt('TokenEndpoint').toString();
    }
    /**
       * logout url
       */
    get logoutEndpoint() {
        return super.getAtt('LogoutEndpoint').toString();
    }
    /**
       * user-info endpoint
       */
    get userInfoEndpoint() {
        return super.getAtt('UserInfoEndpoint').toString();
    }
}
exports.CfnLoginClient = CfnLoginClient;
