// @vw-sre/vws-cdk/lib/vws/cfn-rvs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnRvsSendTransfer = exports.CfnRvsReceiveRule = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const vdsnPattern = (minSuffixLength = 1) => new RegExp(`^[QED][A-Z0-9]{2}.[A-Z0-9]{6}.[.\\-/()&A-Z0-9]{${minSuffixLength},15}$`);
const vdsnRegex = vdsnPattern();
const vdsnPrefixRegex = vdsnPattern(5);
const validateVdsn = (vdsn) => {
    const match = vdsn.match(vdsnRegex);
    if (!match) {
        throw new Error(`validation error: VDSN ${vdsn} does not match the pattern ${vdsnRegex.source}`);
    }
};
const validateVdsnPrefix = (vdsnPrefix) => {
    const match = vdsnPrefix.match(vdsnPrefixRegex);
    if (!match) {
        throw new Error(`validation error: VDSN ${vdsnPrefix} does not match the pattern ${vdsnPrefixRegex.source}`);
    }
};
/**
 * VWS RVS Receive Rule.
 * This creates a VWS::RVS::ReceiveRule resource.
 * This resource delivers objects it has received on its RVS station to your s3 bucket.
 *
 * @deprecated A custom L2 construct will be provided soon
 * @see https://github.com/vw-sre/vws-cdk/issues/410
 */
class CfnRvsReceiveRule extends aws_cdk_lib_1.CfnResource {
    constructor(scope, id, properties) {
        CfnRvsReceiveRule.validateTransfer(properties.transfer);
        super(scope, id, {
            type: 'VWS::RVS::ReceiveRule',
            properties: {
                Transfer: {
                    VDSN: properties.transfer.vdsn,
                    VDSNPrefix: properties.transfer.vdsnPrefix,
                },
                CharsetConversion: properties.charsetConversion,
                Destination: {
                    S3BucketName: properties.destination.s3BucketName,
                    S3BucketPrefix: properties.destination.s3BucketPrefix,
                    RoleArn: properties.destination.roleArn,
                },
            },
        });
    }
}
exports.CfnRvsReceiveRule = CfnRvsReceiveRule;
CfnRvsReceiveRule.validateTransfer = (transfer) => {
    if (!transfer.vdsn && !transfer.vdsnPrefix) {
        throw new Error('validation error: Either vdsn or vdsnPrefix has to be set.');
    }
    if (transfer.vdsn && transfer.vdsnPrefix) {
        throw new Error('validation error: You cannot specify both vdsn and vdsnPrefix.');
    }
    if (transfer.vdsn) {
        validateVdsn(transfer.vdsn);
    }
    if (transfer.vdsnPrefix) {
        validateVdsnPrefix(transfer.vdsnPrefix);
    }
};
/**
 * VWS RVS SendTransfer
 * This creates a VWS::RVS::SendTransfer resource which allows you to transfer objects from an S3 bucket via RVS.
 *
 * @deprecated A custom L2 construct will be provided soon
 * @see https://github.com/vw-sre/vws-cdk/issues/410
 */
class CfnRvsSendTransfer extends aws_cdk_lib_1.CfnResource {
    constructor(scope, id, properties) {
        validateVdsn(properties.transfer.vdsn);
        super(scope, id, {
            type: 'VWS::RVS::SendTransfer',
            properties: {
                Source: {
                    S3BucketName: properties.source.s3BucketName,
                    S3ObjectKey: properties.source.s3ObjectKey,
                    RoleArn: properties.source.roleArn,
                },
                Transfer: {
                    VDSN: properties.transfer.vdsn,
                },
                CharsetConversion: properties.charsetConversion,
                Encryption: properties.encryption,
            },
        });
    }
    /**
       * The Amazon Resource Name (ARN) of the EventBridge Event Bus where to forward S3 events to.
       */
    get eventBusArn() {
        return super.getAtt('EventBusArn').toString();
    }
}
exports.CfnRvsSendTransfer = CfnRvsSendTransfer;
