// @vw-sre/vws-cdk/lib/vws/cfn-proxy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AllowedSuffixes = exports.CfnProxyCredentials = exports.CfnProxy = void 0;
/**
 * @field allowedSuffixes: list of allowed DNS suffixes
 * @field allowedCidrs: list of allowed target CIDR ranges
 * @field allowedPorts: list of ports allowed to connect to
 */
const aws_cdk_lib_1 = require("aws-cdk-lib");
/**
 * VWS proxy service instance. Provides a VPC endpoint service name after creation.
 *
 * @deprecated A custom L2 construct will be provided soon
 * @see https://github.com/vw-sre/vws-cdk/issues/409
 */
class CfnProxy extends aws_cdk_lib_1.CfnResource {
    constructor(scope, id, properties) {
        if (properties.allowedPorts.length == 0 || (properties.allowedCidrs.length == 0 && properties.allowedSuffixes.length == 0)) {
            throw new Error('validation error: either cidrs and/or dns suffixes and ports required');
        }
        super(scope, id, {
            type: 'VWS::Proxy::Instance',
            properties: {
                AllowedSuffixes: aws_cdk_lib_1.Lazy.list({ produce: () => properties.allowedSuffixes }, { omitEmpty: true }),
                AllowedCidrs: aws_cdk_lib_1.Lazy.list({ produce: () => properties.allowedCidrs }, { omitEmpty: true }),
                AllowedPorts: aws_cdk_lib_1.Lazy.list({ produce: () => properties.allowedPorts.map(it => `${it}`) }, { omitEmpty: true }),
            },
        });
    }
    /**
       * returns proxy VPC endpoint service name
       */
    get serviceName() {
        return super.getAtt('ServiceName').toString();
    }
    /**
       * returns private proxy endpoint dns name
       */
    get dnsName() {
        return super.getAtt('DnsName').toString();
    }
}
exports.CfnProxy = CfnProxy;
/**
 * Generates a secret/password for a proxy user and stores it in AWS SecretsManager.
 *
 * In order to be able to access the secret a principal need the following permission in their policy:
 - Effect: Allow
 Action:
 - kms:Decrypt
 - kms:DescribeKey
 Resource: !Sub 'arn:aws:kms:${AWS::Region}:637378239786:key/*'

 @deprecated A custom L2 construct will be provided soon
 @see https://github.com/vw-sre/vws-cdk/issues/409
 */
class CfnProxyCredentials extends aws_cdk_lib_1.CfnResource {
    constructor(scope, id, properties) {
        super(scope, id, {
            type: 'VWS::Proxy::Credentials',
            properties: {
                Principals: properties.principals,
                Instance: { Ref: properties.instance.logicalId },
            },
        });
    }
    /**
       * id returns the proxy credential id
       */
    get id() {
        return super.getAtt('Id').toString();
    }
    /**
       * secretsArn returns the AWS SecretsManager Arn for the secret containing the proxy password
       */
    get secretsArn() {
        return super.getAtt('SecretArn').toString();
    }
}
exports.CfnProxyCredentials = CfnProxyCredentials;
/**
 * Collection of default suffixes
 *
 * @deprecated
 */
class AllowedSuffixes {
}
exports.AllowedSuffixes = AllowedSuffixes;
AllowedSuffixes.ALPINE_PACKAGES = ['alpine.global.ssl.fastly.net', 'dl-cdn.alpinelinux.org'];
AllowedSuffixes.DOCKER = ['docker.io', 'docker.com'];
AllowedSuffixes.GITHUB = ['github.com', 'githubusercontent.com', 'ghcr.io'];
AllowedSuffixes.NEW_RELIC = ['newrelic.com', 'newrelic.eu'];
AllowedSuffixes.NPM = ['npmjs.org', 'yarnpkg.com'];
AllowedSuffixes.PYTHON = ['pypi.python.org', 'pypi.org', 'files.pythonhosted.org'];
