// @vw-sre/vws-cdk/lib/vws/connectivity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectivityVPC = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_ec2_1 = require("aws-cdk-lib/aws-ec2");
const aws_route53resolver_1 = require("aws-cdk-lib/aws-route53resolver");
const constructs_1 = require("constructs");
const aspect_vpc_1 = require("../aspects/aspect-vpc");
const generated_1 = require("../generated");
/**
 * This Contruct provides a VPC with public and isolated subnets with cidr: 10.0.0.0/16.
 * In addition the VPC contains subnets within a VWS::Network cidr range and a transit gateway connection.
 *
 * To deploy your components into specific subnets please use the exposed subnet
 * lists instead accessing them via the exposed vpc!
 * @experimental
 */
class ConnectivityVPC extends constructs_1.Construct {
    constructor(scope, id, props) {
        super(scope, id);
        if (props.azs.length < 1) {
            throw new Error(`Invalid property azs: "${JSON.stringify(props.azs)}" given. Make sure to provide at least one availability zone.`);
        }
        const cfnNetwork = new generated_1.CfnNetwork(this, 'Network', {
            pool: props.pool,
            size: props.size,
        });
        const stockVpc = new aws_ec2_1.Vpc(this, 'VPC', {
            cidr: '10.0.0.0/16',
            subnetConfiguration: [
                {
                    subnetType: aws_ec2_1.SubnetType.PUBLIC,
                    cidrMask: 24,
                    name: 'Ingress',
                    reserved: false,
                },
                {
                    subnetType: aws_ec2_1.SubnetType.PRIVATE_ISOLATED,
                    cidrMask: 24,
                    name: 'App',
                    reserved: false,
                },
            ],
            maxAzs: props.azs.length,
        });
        aws_cdk_lib_1.Aspects.of(stockVpc).add(new aspect_vpc_1.DisableMapPublicIpOnLaunch());
        const cidrBlock = new aws_ec2_1.CfnVPCCidrBlock(this, 'NetworkCidr', {
            vpcId: stockVpc.vpcId,
            cidrBlock: cfnNetwork.attrNetwork,
        });
        let cidrBits = 32 - 1 - props.size;
        let cidrCount = props.azs.length;
        const cfnConnectivitySubnets = [];
        props.azs.forEach((az, index) => {
            cfnConnectivitySubnets.push(new aws_ec2_1.CfnSubnet(this, `ConnectivitySubnet${index}`, {
                vpcId: stockVpc.vpcId,
                cidrBlock: aws_cdk_lib_1.Fn.select(index, aws_cdk_lib_1.Fn.cidr(cfnNetwork.attrNetwork, cidrCount, String(cidrBits))),
                availabilityZone: az,
                mapPublicIpOnLaunch: false,
            }));
        });
        cfnConnectivitySubnets.forEach((subnet) => {
            subnet.addDependsOn(cfnNetwork);
            subnet.addDependsOn(cidrBlock);
        });
        const cfnRouteTable = new aws_ec2_1.CfnRouteTable(this, 'RouteTable', {
            vpcId: stockVpc.vpcId,
        });
        const cfnAttachment = new aws_ec2_1.CfnTransitGatewayAttachment(this, 'Attachment', {
            vpcId: stockVpc.vpcId,
            subnetIds: cfnConnectivitySubnets.map(s => s.ref),
            transitGatewayId: cfnNetwork.attrTransitGatewayId,
        });
        const cfnDefaultRoute = new aws_ec2_1.CfnRoute(this, 'DefaultRoute', {
            destinationCidrBlock: '0.0.0.0/0',
            transitGatewayId: cfnNetwork.attrTransitGatewayId,
            routeTableId: cfnRouteTable.ref,
        });
        cfnDefaultRoute.addDependsOn(cfnAttachment);
        cfnConnectivitySubnets.map((subnet, index) => new aws_ec2_1.CfnSubnetRouteTableAssociation(this, `Table${index}`, {
            routeTableId: cfnRouteTable.ref,
            subnetId: subnet.ref,
        }));
        new generated_1.CfnConnection(this, 'Connection', {
            vpcId: stockVpc.vpcId,
            network: cfnNetwork.ref,
            transitGatewayId: cfnNetwork.attrTransitGatewayId,
        });
        if (props.dnsResolvers) {
            const resolver = new generated_1.CfnResolver(this, 'IntranetResolver', {
                pool: 'intranet',
            });
            new aws_route53resolver_1.CfnResolverRuleAssociation(this, 'RRA0', {
                vpcId: stockVpc.vpcId,
                resolverRuleId: aws_cdk_lib_1.Fn.select(0, resolver.attrRules),
            });
            new aws_route53resolver_1.CfnResolverRuleAssociation(this, 'RRA1', {
                vpcId: stockVpc.vpcId,
                resolverRuleId: aws_cdk_lib_1.Fn.select(1, resolver.attrRules),
            });
            new aws_route53resolver_1.CfnResolverRuleAssociation(this, 'RRA2', {
                vpcId: stockVpc.vpcId,
                resolverRuleId: aws_cdk_lib_1.Fn.select(2, resolver.attrRules),
            });
        }
        this.vpc = aws_ec2_1.Vpc.fromVpcAttributes(this, 'VpcFromAttrs', {
            vpcId: stockVpc.vpcId,
            availabilityZones: props.azs,
            publicSubnetIds: stockVpc.publicSubnets.map(subnet => subnet.subnetId),
            isolatedSubnetIds: stockVpc.isolatedSubnets.map(subnet => subnet.subnetId),
            privateSubnetIds: cfnConnectivitySubnets.map(subnet => subnet.ref),
        });
        this.ingressSubnets = this.vpc.publicSubnets;
        this.appSubnets = this.vpc.isolatedSubnets;
        this.connectivitySubnets = this.vpc.privateSubnets;
    }
}
exports.ConnectivityVPC = ConnectivityVPC;
