// @vw-sre/vws-cdk/lib/vws/domain.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Domain = void 0;
const aws_route53_1 = require("aws-cdk-lib/aws-route53");
const constructs_1 = require("constructs");
const generated_1 = require("../generated");
const cfn_domain_1 = require("./cfn-domain");
class Domain extends constructs_1.Construct {
    constructor(scope, id, props) {
        super(scope, id);
        const domain = props ?
            new cfn_domain_1.CfnDomain(this, 'Domain', { name: props.domain }) :
            new cfn_domain_1.CfnDomain(this, 'Domain');
        this.name = domain.name;
        this.hostedZone = new aws_route53_1.HostedZone(this, 'HostedZone', {
            zoneName: domain.name,
            comment: `Hosted zone ${domain.logicalId}`,
        });
        const domainDelegation = new generated_1.CfnDelegation(this, 'DomainDelegation', {
            domain: domain.name,
            nameservers: this.hostedZone.hostedZoneNameServers,
        });
        domainDelegation.addDependsOn(domain);
    }
}
exports.Domain = Domain;
