// @vw-sre/vws-cdk/lib/vws/outgoing-proxy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OutgoingProxyCredentials = exports.OutgoingProxy = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_iam_1 = require("aws-cdk-lib/aws-iam");
const aws_kms_1 = require("aws-cdk-lib/aws-kms");
const aws_secretsmanager_1 = require("aws-cdk-lib/aws-secretsmanager");
const generated_1 = require("../generated");
/**
 * Creates VWS::Proxy::Instance
 *
 * @see https://docs.vwapps.cloud/vws-vanguard/services/resources/proxy/
 */
class OutgoingProxy extends aws_cdk_lib_1.Resource {
    constructor(scope, id, props) {
        super(scope, id);
        this.resource = new generated_1.CfnInstance(this, 'Resource', {
            allowedCidrs: props.allowedCidrs,
            allowedPorts: props.allowedPorts,
            allowedSuffixes: props.allowedSuffixes,
        });
    }
    get id() {
        return this.resource.attrId;
    }
    get serviceName() {
        return this.resource.attrServiceName;
    }
    get dnsName() {
        return this.resource.attrDnsName;
    }
}
exports.OutgoingProxy = OutgoingProxy;
/**
 * Creates VWS::Proxy::Credentials and grants access to them for all iam.IGrantable principals
 *
 * @see https://docs.vwapps.cloud/vws-vanguard/services/resources/proxy/
 */
class OutgoingProxyCredentials extends aws_cdk_lib_1.Resource {
    constructor(scope, id, props) {
        super(scope, id);
        this.resource = new generated_1.CfnCredentials(this, 'Resource', {
            instance: props.instance.id,
            principals: props.principals.map(x => {
                if (x instanceof aws_iam_1.AccountPrincipal) {
                    return x.arn;
                }
                else {
                    return x.roleArn;
                }
            }),
        });
        for (let principal of props.principals) {
            if (!(principal instanceof aws_iam_1.AccountPrincipal)) {
                this.grantRead(principal);
            }
        }
        this.secret = aws_secretsmanager_1.Secret.fromSecretPartialArn(this, 'Secret', this.resource.attrSecretArn);
        this.kmsKey = aws_kms_1.Key.fromKeyArn(this, 'Key', this.resource.attrKmsKeyArn);
    }
    grantRead(identity) {
        aws_iam_1.Grant.addToPrincipal({
            grantee: identity,
            actions: [
                'secretsmanager:GetResourcePolicy',
                'secretsmanager:GetSecretValue',
                'secretsmanager:DescribeSecret',
                'secretsmanager:ListSecretVersionIds',
            ],
            resourceArns: [this.resource.attrSecretArn],
        });
        aws_iam_1.Grant.addToPrincipal({
            grantee: identity,
            actions: [
                'kms:Decrypt',
                'kms:DescribeKey',
            ],
            resourceArns: [this.resource.attrKmsKeyArn],
        });
    }
}
exports.OutgoingProxyCredentials = OutgoingProxyCredentials;
