// @vw-sre/vws-cdk/lib/vws/rvs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RvsSendTransfer = exports.RvsReceiveRule = exports.RvsCharsetConversion = void 0;
const path = require("path");
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_events_1 = require("aws-cdk-lib/aws-events");
const aws_iam_1 = require("aws-cdk-lib/aws-iam");
const generated_1 = require("../generated");
const vdsnPattern = (minSuffixLength = 1) => new RegExp(`^[QED][A-Z0-9]{2}.[A-Z0-9]{6}.[.\\-/()&A-Z0-9]{${minSuffixLength},15}$`);
const vdsnRegex = vdsnPattern();
const vdsnPrefixRegex = vdsnPattern(5);
const validateVdsn = (vdsn) => {
    const match = vdsn.match(vdsnRegex);
    if (!match) {
        throw new Error(`validation error: VDSN ${vdsn} does not match the pattern ${vdsnRegex.source}`);
    }
};
const validateVdsnPrefix = (vdsnPrefix) => {
    const match = vdsnPrefix.match(vdsnPrefixRegex);
    if (!match) {
        throw new Error(`validation error: VDSN ${vdsnPrefix} does not match the pattern ${vdsnPrefixRegex.source}`);
    }
};
var RvsCharsetConversion;
(function (RvsCharsetConversion) {
    RvsCharsetConversion["ANSI_IBM037"] = "ANSI-IBM037";
    RvsCharsetConversion["ANSI_IBM273"] = "ANSI-IBM273";
    RvsCharsetConversion["ASCII_IBM037"] = "ASCII-IBM037";
    RvsCharsetConversion["ASCII_IBM273"] = "ASCII-IBM273";
    RvsCharsetConversion["ASCII_IBM500"] = "ASCII-IBM500";
    RvsCharsetConversion["IBM037_ANSI"] = "IBM037-ANSI";
    RvsCharsetConversion["IBM037_ASCII"] = "IBM037-ASCII";
    RvsCharsetConversion["IBM273_ANSI"] = "IBM273-ANSI";
    RvsCharsetConversion["IBM273_ASCII"] = "IBM273-ASCII";
    RvsCharsetConversion["IBM500_ASCII"] = "IBM500-ASCII";
})(RvsCharsetConversion = exports.RvsCharsetConversion || (exports.RvsCharsetConversion = {}));
class RvsReceiveRule extends aws_cdk_lib_1.Resource {
    constructor(scope, id, props) {
        RvsReceiveRule.validateTransfer(props.transfer);
        super(scope, id);
        const role = this.setupRole(props);
        new generated_1.CfnReceiveRule(this, 'Resource', {
            transfer: {
                vdsn: props.transfer.vdsn,
                vdsnPrefix: props.transfer.vdsnPrefix,
            },
            charsetConversion: RvsReceiveRule.convertReceiveCharset(props.charsetConversion),
            destination: {
                s3BucketName: props.destination.bucket.bucketName,
                s3BucketPrefix: props.destination.s3BucketPrefix,
                roleArn: role.roleArn,
            },
        });
    }
    static convertReceiveCharset(charset) {
        switch (charset) {
            case RvsCharsetConversion.ANSI_IBM037:
                return generated_1.CfnReceiveRulePropsCharsetConversion.ANSI_IBM037;
            case RvsCharsetConversion.ANSI_IBM273:
                return generated_1.CfnReceiveRulePropsCharsetConversion.ANSI_IBM273;
            case RvsCharsetConversion.ASCII_IBM037:
                return generated_1.CfnReceiveRulePropsCharsetConversion.ASCII_IBM037;
            case RvsCharsetConversion.ASCII_IBM273:
                return generated_1.CfnReceiveRulePropsCharsetConversion.ASCII_IBM273;
            case RvsCharsetConversion.ASCII_IBM500:
                return generated_1.CfnReceiveRulePropsCharsetConversion.ASCII_IBM500;
            case RvsCharsetConversion.IBM037_ANSI:
                return generated_1.CfnReceiveRulePropsCharsetConversion.IBM037_ANSI;
            case RvsCharsetConversion.IBM037_ASCII:
                return generated_1.CfnReceiveRulePropsCharsetConversion.IBM037_ASCII;
            case RvsCharsetConversion.IBM273_ANSI:
                return generated_1.CfnReceiveRulePropsCharsetConversion.IBM273_ANSI;
            case RvsCharsetConversion.IBM273_ASCII:
                return generated_1.CfnReceiveRulePropsCharsetConversion.IBM273_ASCII;
            case RvsCharsetConversion.IBM500_ASCII:
                return generated_1.CfnReceiveRulePropsCharsetConversion.IBM500_ASCII;
            case undefined:
                return undefined;
        }
    }
    setupRole(props) {
        var _a, _b;
        if (props.destination.role === undefined) {
            props.destination.role = new aws_iam_1.Role(this, 'ResourceRole', {
                assumedBy: new aws_iam_1.AccountPrincipal('074430740969'),
            });
        }
        else {
            props.destination.role.grantAssumeRole(new aws_iam_1.AccountPrincipal('074430740969'));
        }
        const allObjects = path.join((_a = props.destination.s3BucketPrefix) !== null && _a !== void 0 ? _a : '', '*');
        props.destination.role.addToPrincipalPolicy(new aws_iam_1.PolicyStatement({
            effect: aws_iam_1.Effect.ALLOW,
            actions: [
                's3:PutObject',
                's3:GetBucketLocation',
            ],
            resources: [
                props.destination.bucket.bucketArn,
                props.destination.bucket.arnForObjects(allObjects),
            ],
        }));
        (_b = props.destination.bucket.encryptionKey) === null || _b === void 0 ? void 0 : _b.grantEncryptDecrypt(props.destination.role);
        return props.destination.role;
    }
}
exports.RvsReceiveRule = RvsReceiveRule;
RvsReceiveRule.validateTransfer = (transfer) => {
    if (!transfer.vdsn && !transfer.vdsnPrefix) {
        throw new Error('validation error: Either vdsn or vdsnPrefix has to be set.');
    }
    if (transfer.vdsn && transfer.vdsnPrefix) {
        throw new Error('validation error: You cannot specify both vdsn and vdsnPrefix.');
    }
    if (transfer.vdsn) {
        validateVdsn(transfer.vdsn);
    }
    if (transfer.vdsnPrefix) {
        validateVdsnPrefix(transfer.vdsnPrefix);
    }
};
class RvsSendTransfer extends aws_cdk_lib_1.Resource {
    constructor(scope, id, props) {
        validateVdsn(props.transfer.vdsn);
        super(scope, id);
        const role = this.setupRole(props);
        const resource = new generated_1.CfnSendTransfer(this, 'Resource', {
            source: {
                s3BucketName: props.source.bucket.bucketName,
                s3ObjectKey: props.source.objectKey,
                roleArn: role.roleArn,
            },
            transfer: {
                vdsn: props.transfer.vdsn,
            },
            charsetConversion: RvsSendTransfer.convertSendCharset(props.charsetConversion),
            encryption: props.encryption,
        });
        this.eventBus = aws_events_1.EventBus.fromEventBusArn(this, 'EventBus', resource.attrEventBusArn);
    }
    static convertSendCharset(charset) {
        switch (charset) {
            case RvsCharsetConversion.ANSI_IBM037:
                return generated_1.CfnSendTransferPropsCharsetConversion.ANSI_IBM037;
            case RvsCharsetConversion.ANSI_IBM273:
                return generated_1.CfnSendTransferPropsCharsetConversion.ANSI_IBM273;
            case RvsCharsetConversion.ASCII_IBM037:
                return generated_1.CfnSendTransferPropsCharsetConversion.ASCII_IBM037;
            case RvsCharsetConversion.ASCII_IBM273:
                return generated_1.CfnSendTransferPropsCharsetConversion.ASCII_IBM273;
            case RvsCharsetConversion.ASCII_IBM500:
                return generated_1.CfnSendTransferPropsCharsetConversion.ASCII_IBM500;
            case RvsCharsetConversion.IBM037_ANSI:
                return generated_1.CfnSendTransferPropsCharsetConversion.IBM037_ANSI;
            case RvsCharsetConversion.IBM037_ASCII:
                return generated_1.CfnSendTransferPropsCharsetConversion.IBM037_ASCII;
            case RvsCharsetConversion.IBM273_ANSI:
                return generated_1.CfnSendTransferPropsCharsetConversion.IBM273_ANSI;
            case RvsCharsetConversion.IBM273_ASCII:
                return generated_1.CfnSendTransferPropsCharsetConversion.IBM273_ASCII;
            case RvsCharsetConversion.IBM500_ASCII:
                return generated_1.CfnSendTransferPropsCharsetConversion.IBM500_ASCII;
            case undefined:
                return undefined;
        }
    }
    setupRole(props) {
        var _a;
        if (props.source.role === undefined) {
            props.source.role = new aws_iam_1.Role(this, 'ResourceRole', {
                assumedBy: new aws_iam_1.AccountPrincipal('074430740969'),
            });
        }
        else {
            props.source.role.grantAssumeRole(new aws_iam_1.AccountPrincipal('074430740969'));
        }
        props.source.bucket.grantRead(props.source.role, props.source.objectKey);
        (_a = props.source.bucket.encryptionKey) === null || _a === void 0 ? void 0 : _a.grantDecrypt(props.source.role);
        return props.source.role;
    }
}
exports.RvsSendTransfer = RvsSendTransfer;
