// @vw-sre/vws-cdk/lib/vws/shared-vpc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharedVpc = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_ec2_1 = require("aws-cdk-lib/aws-ec2");
const generated_1 = require("../generated");
class SharedVpc extends aws_cdk_lib_1.Resource {
    constructor(scope, id, props) {
        var _a;
        super(scope, id, props);
        const resource = new generated_1.CfnShared(this, 'Resource', {});
        this.vpc = SharedVpc.from(resource, (_a = props === null || props === void 0 ? void 0 : props.availabilityZones) !== null && _a !== void 0 ? _a : this.stack.availabilityZones);
    }
    static from(sharedVpc, availabilityZones) {
        let subnetIds = [];
        availabilityZones.forEach((_, index) => {
            subnetIds.push(aws_cdk_lib_1.Fn.select(index, sharedVpc.attrSubnets));
        });
        return aws_ec2_1.Vpc.fromVpcAttributes(sharedVpc, `${sharedVpc.attrId}-Vpc`, {
            vpcId: sharedVpc.attrVpcId,
            availabilityZones: availabilityZones,
            isolatedSubnetIds: subnetIds,
        });
    }
}
exports.SharedVpc = SharedVpc;
