// @vw-sre/vws-cdk/lib/generated/vws/login/vws-login-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnClient = exports.toJson_CfnClientProps = void 0;
// Generated by cdk-import
const cdk = require("aws-cdk-lib");
/**
 * Converts an object of type 'CfnClientProps' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnClientProps(obj) {
    var _a, _b;
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'CallbackUrl': (_a = obj.callbackUrl) === null || _a === void 0 ? void 0 : _a.map(y => y),
        'LogoutUrl': (_b = obj.logoutUrl) === null || _b === void 0 ? void 0 : _b.map(y => y),
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnClientProps = toJson_CfnClientProps;
/* eslint-enable max-len, quote-props */
/**
 * A CloudFormation `VWS::Login::Client`
 *
 * @cloudformationResource VWS::Login::Client
 * @stability external
 */
class CfnClient extends cdk.CfnResource {
    /**
     * Create a new `VWS::Login::Client`.
     *
     * @param scope - scope in which this resource is defined
     * @param id    - scoped id of the resource
     * @param props - resource properties
     */
    constructor(scope, id, props) {
        super(scope, id, { type: CfnClient.CFN_RESOURCE_TYPE_NAME, properties: toJson_CfnClientProps(props) });
        this.props = props;
        this.attrId = cdk.Token.asString(this.getAtt('Id'));
        this.attrSecretArn = cdk.Token.asString(this.getAtt('SecretArn'));
        this.attrIssuer = cdk.Token.asString(this.getAtt('Issuer'));
        this.attrAuthorizationEndpoint = cdk.Token.asString(this.getAtt('AuthorizationEndpoint'));
        this.attrTokenEndpoint = cdk.Token.asString(this.getAtt('TokenEndpoint'));
        this.attrLogoutEndpoint = cdk.Token.asString(this.getAtt('LogoutEndpoint'));
        this.attrUserInfoEndpoint = cdk.Token.asString(this.getAtt('UserInfoEndpoint'));
    }
}
exports.CfnClient = CfnClient;
/**
* The CloudFormation resource type name for this resource class.
*/
CfnClient.CFN_RESOURCE_TYPE_NAME = "VWS::Login::Client";
