// @vw-sre/vws-cdk/lib/generated/vws/proxy/vws-proxy-instance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnInstance = exports.toJson_CfnInstanceProps = void 0;
// Generated by cdk-import
const cdk = require("aws-cdk-lib");
/**
 * Converts an object of type 'CfnInstanceProps' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnInstanceProps(obj) {
    var _a, _b, _c;
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'AllowedSuffixes': (_a = obj.allowedSuffixes) === null || _a === void 0 ? void 0 : _a.map(y => y),
        'AllowedCidrs': (_b = obj.allowedCidrs) === null || _b === void 0 ? void 0 : _b.map(y => y),
        'AllowedPorts': (_c = obj.allowedPorts) === null || _c === void 0 ? void 0 : _c.map(y => y),
        'Environment': obj.environment,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnInstanceProps = toJson_CfnInstanceProps;
/* eslint-enable max-len, quote-props */
/**
 * A CloudFormation `VWS::Proxy::Instance`
 *
 * @cloudformationResource VWS::Proxy::Instance
 * @stability external
 */
class CfnInstance extends cdk.CfnResource {
    /**
     * Create a new `VWS::Proxy::Instance`.
     *
     * @param scope - scope in which this resource is defined
     * @param id    - scoped id of the resource
     * @param props - resource properties
     */
    constructor(scope, id, props) {
        super(scope, id, { type: CfnInstance.CFN_RESOURCE_TYPE_NAME, properties: toJson_CfnInstanceProps(props) });
        this.props = props;
        this.attrId = cdk.Token.asString(this.getAtt('Id'));
        this.attrServiceName = cdk.Token.asString(this.getAtt('ServiceName'));
        this.attrDnsName = cdk.Token.asString(this.getAtt('DnsName'));
    }
}
exports.CfnInstance = CfnInstance;
/**
* The CloudFormation resource type name for this resource class.
*/
CfnInstance.CFN_RESOURCE_TYPE_NAME = "VWS::Proxy::Instance";
