// @vw-sre/vws-cdk/lib/generated/vws/proxy/vws-proxy-credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnCredentials = exports.toJson_CfnCredentialsProps = void 0;
// Generated by cdk-import
const cdk = require("aws-cdk-lib");
/**
 * Converts an object of type 'CfnCredentialsProps' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnCredentialsProps(obj) {
    var _a;
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Instance': obj.instance,
        'Principals': (_a = obj.principals) === null || _a === void 0 ? void 0 : _a.map(y => y),
        'Environment': obj.environment,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnCredentialsProps = toJson_CfnCredentialsProps;
/* eslint-enable max-len, quote-props */
/**
 * A CloudFormation `VWS::Proxy::Credentials`
 *
 * @cloudformationResource VWS::Proxy::Credentials
 * @stability external
 */
class CfnCredentials extends cdk.CfnResource {
    /**
     * Create a new `VWS::Proxy::Credentials`.
     *
     * @param scope - scope in which this resource is defined
     * @param id    - scoped id of the resource
     * @param props - resource properties
     */
    constructor(scope, id, props) {
        super(scope, id, { type: CfnCredentials.CFN_RESOURCE_TYPE_NAME, properties: toJson_CfnCredentialsProps(props) });
        this.props = props;
        this.attrId = cdk.Token.asString(this.getAtt('Id'));
        this.attrSecretArn = cdk.Token.asString(this.getAtt('SecretArn'));
        this.attrKmsKeyArn = cdk.Token.asString(this.getAtt('KmsKeyArn'));
    }
}
exports.CfnCredentials = CfnCredentials;
/**
* The CloudFormation resource type name for this resource class.
*/
CfnCredentials.CFN_RESOURCE_TYPE_NAME = "VWS::Proxy::Credentials";
