// @vw-sre/vws-cdk/lib/aspects/aspect-lambda.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LambdaVPCAspect = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_iam_1 = require("aws-cdk-lib/aws-iam");
const aws_lambda_1 = require("aws-cdk-lib/aws-lambda");
/**
 * usage: Aspects.of(scope).add(new LambdaVPCAspect(vpc, sg))
 *  @param vpc VPC to add to all lambda functions that are not bound to a vpc. <br/>
 *       <i>Use a CfnSharedVpc#vpn for this, in order to use the CDK custom resources like BucketDeployment</i>
 *  @param securityGroup security group all the lambdas functions will be assigned to
 */
class LambdaVPCAspect {
    constructor(vpc, securityGroup, subnetSelection) {
        this.vpc = vpc;
        this.securityGroup = securityGroup;
        this.subnetSelection = subnetSelection;
    }
    /**
     * All aspects can visit an IConstruct.
     */
    visit(node) {
        this.patchRole(node);
        this.patchFunction(node);
    }
    patchFunction(node) {
        if (aws_cdk_lib_1.CfnResource.isCfnResource(node) && node.cfnResourceType == 'AWS::Lambda::Function') {
            if (node instanceof aws_lambda_1.CfnFunction) {
                if (!node.vpcConfig) {
                    node.addPropertyOverride('VpcConfig', this.getVpcConfig());
                }
            }
            else {
                // @ts-ignore
                if (!node.cfnProperties.VpcConfig) {
                    // @ts-ignore
                    node._cfnProperties = Object.assign(Object.assign({}, node.cfnProperties), { VpcConfig: this.getVpcConfig() });
                }
            }
        }
    }
    patchRole(node) {
        var _a;
        if (node instanceof aws_iam_1.Role) {
            if (node.managedPolicies
                .find((it) => it.managedPolicyArn.endsWith(LambdaVPCAspect.LAMBDA_BASIC_MANAGED_POLICY_NAME))) {
                node.addManagedPolicy(aws_iam_1.ManagedPolicy.fromAwsManagedPolicyName(LambdaVPCAspect.LAMBDA_VPC_MANAGED_POLICY_NAME));
            }
        }
        else if (aws_cdk_lib_1.CfnResource.isCfnResource(node) && node.cfnResourceType == aws_iam_1.CfnRole.CFN_RESOURCE_TYPE_NAME) {
            const oldPolicies = node._cfnProperties.ManagedPolicyArns || [];
            const basicManagedPolicy = (_a = oldPolicies
                .map((it) => this.mapToString(it))
                .find(((it) => it.endsWith(LambdaVPCAspect.LAMBDA_BASIC_MANAGED_POLICY_NAME)))) === null || _a === void 0 ? void 0 : _a[0];
            if (basicManagedPolicy) {
                node.addPropertyOverride('ManagedPolicyArns', [
                    ...oldPolicies,
                    { 'Fn::Join': ['', ['arn:', { Ref: 'AWS::Partition' }, `:iam::aws:policy/${LambdaVPCAspect.LAMBDA_VPC_MANAGED_POLICY_NAME}`]] },
                ]);
            }
        }
    }
    getVpcConfig() {
        const { subnetIds } = this.vpc.selectSubnets(this.subnetSelection);
        return {
            SecurityGroupIds: [this.securityGroup].map(sg => sg.securityGroupId),
            SubnetIds: subnetIds,
        };
    }
    mapToString(it) {
        if (typeof it === 'object') {
            return Object.values(it)[0] || '';
        }
        else {
            return it;
        }
    }
}
exports.LambdaVPCAspect = LambdaVPCAspect;
LambdaVPCAspect.LAMBDA_VPC_MANAGED_POLICY_NAME = 'service-role/AWSLambdaVPCAccessExecutionRole';
LambdaVPCAspect.LAMBDA_BASIC_MANAGED_POLICY_NAME = 'service-role/AWSLambdaBasicExecutionRole';
