// @vw-sre/vws-cdk/lib/generated/vws/connectivity/vws-connectivity-network.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CfnNetwork = exports.toJson_CfnNetworkProps = void 0;
// Generated by cdk-import
const cdk = require("aws-cdk-lib");
/**
 * Converts an object of type 'CfnNetworkProps' to JSON representation.
 */
/* eslint-disable max-len, quote-props */
function toJson_CfnNetworkProps(obj) {
    if (obj === undefined) {
        return undefined;
    }
    const result = {
        'Size': obj.size,
        'Pool': obj.pool,
    };
    // filter undefined values
    return Object.entries(result).reduce((r, i) => (i[1] === undefined) ? r : (Object.assign(Object.assign({}, r), { [i[0]]: i[1] })), {});
}
exports.toJson_CfnNetworkProps = toJson_CfnNetworkProps;
/* eslint-enable max-len, quote-props */
/**
 * A CloudFormation `VWS::Connectivity::Network`
 *
 * @cloudformationResource VWS::Connectivity::Network
 * @stability external
 */
class CfnNetwork extends cdk.CfnResource {
    /**
     * Create a new `VWS::Connectivity::Network`.
     *
     * @param scope - scope in which this resource is defined
     * @param id    - scoped id of the resource
     * @param props - resource properties
     */
    constructor(scope, id, props) {
        super(scope, id, { type: CfnNetwork.CFN_RESOURCE_TYPE_NAME, properties: toJson_CfnNetworkProps(props) });
        this.props = props;
        this.attrId = cdk.Token.asString(this.getAtt('Id'));
        this.attrNetwork = cdk.Token.asString(this.getAtt('Network'));
        this.attrTransitGatewayId = cdk.Token.asString(this.getAtt('TransitGatewayId'));
    }
}
exports.CfnNetwork = CfnNetwork;
/**
* The CloudFormation resource type name for this resource class.
*/
CfnNetwork.CFN_RESOURCE_TYPE_NAME = "VWS::Connectivity::Network";
