// @vw-sre/vws-cdk/lib/stacks/log-group-manager-stack.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogGroupManagerDeploymentPolicy = exports.LogGroupManagerStack = void 0;
const aws_cdk_lib_1 = require("aws-cdk-lib");
const aws_iam_1 = require("aws-cdk-lib/aws-iam");
const aws_sam_1 = require("aws-cdk-lib/aws-sam");
const cfn_shared_vpc_1 = require("../vws/cfn-shared-vpc");
const aws_logs_1 = require("aws-cdk-lib/aws-logs");
class LogGroupManagerStack extends aws_cdk_lib_1.Stack {
    constructor(scope, id, props) {
        var _a, _b;
        super(scope, id, props);
        const vpc = (props === null || props === void 0 ? void 0 : props.vpc) ?
            props.vpc :
            new cfn_shared_vpc_1.CfnSharedVpc(this, 'SharedVpc').vpc(this.availabilityZones);
        if (vpc.isolatedSubnets.length == 0) {
            throw new Error('Vpc needs at least one isolated subnet');
        }
        const regionTable = new aws_cdk_lib_1.CfnMapping(this, 'Regions', {
            mapping: {
                'us-west-2': {
                    id: 'arn:aws:serverlessrepo:us-west-2:637378239786:applications/vws-aws-log-group-manager',
                },
                'us-east-1': {
                    id: 'arn:aws:serverlessrepo:us-east-1:637378239786:applications/vws-aws-log-group-manager',
                },
                'eu-west-1': {
                    id: 'arn:aws:serverlessrepo:eu-west-1:637378239786:applications/vws-aws-log-group-manager',
                },
                'eu-central-1': {
                    id: 'arn:aws:serverlessrepo:eu-central-1:637378239786:applications/vws-aws-log-group-manager',
                },
                'ap-south-1': {
                    id: 'arn:aws:serverlessrepo:ap-south-1:637378239786:applications/vws-aws-log-group-manager',
                },
                'ap-southeast-1': {
                    id: 'arn:aws:serverlessrepo:ap-southeast-1:637378239786:applications/vws-aws-log-group-manager',
                },
                'ap-northeast-2': {
                    id: 'arn:aws:serverlessrepo:ap-northeast-2:637378239786:applications/vws-aws-log-group-manager',
                },
            },
        });
        new aws_sam_1.CfnApplication(this, 'LogGroupManager', {
            location: {
                applicationId: regionTable.findInMap(aws_cdk_lib_1.Aws.REGION, 'id'),
                semanticVersion: '0.1.6',
            },
            parameters: {
                VpcId: vpc.vpcId,
                SubnetId: vpc.isolatedSubnets[0].subnetId,
                LogRetentionPeriod: (_b = (_a = props === null || props === void 0 ? void 0 : props.logRetentionPeriod) === null || _a === void 0 ? void 0 : _a.toString()) !== null && _b !== void 0 ? _b : String(aws_logs_1.RetentionDays.THREE_MONTHS),
            },
        });
    }
}
exports.LogGroupManagerStack = LogGroupManagerStack;
class LogGroupManagerDeploymentPolicy extends aws_iam_1.ManagedPolicy {
    constructor(scope, id, props) {
        const policyProps = {
            managedPolicyName: props.managedPolicyName,
            statements: [
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'ec2:CreateSecurityGroup',
                        'ec2:DescribeSecurityGroups',
                        'ec2:DescribeSubnets',
                        'ec2:DescribeVpcs',
                        'ec2:createTags',
                    ],
                    // no resource restrictions possible
                    resources: ['*'],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'ec2:AuthorizeSecurityGroupEgress',
                        'ec2:DeleteSecurityGroup',
                        'ec2:RevokeSecurityGroupEgress',
                    ],
                    // security group id is not known
                    resources: [`arn:aws:ec2:${aws_cdk_lib_1.Fn.ref('AWS::Region')}:${aws_cdk_lib_1.Fn.ref('AWS::AccountId')}:*`],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'cloudformation:DescribeStacks',
                        'cloudformation:CreateChangeSet',
                        'cloudformation:DescribeChangeSet',
                        'cloudformation:ExecuteChangeSet',
                        'cloudformation:GetTemplateSummary',
                        'cloudformation:DeleteStack',
                        'cloudformation:CreateStack',
                        'cloudformation:UpdateStack',
                    ],
                    resources: [
                        `arn:aws:cloudformation:${aws_cdk_lib_1.Fn.ref('AWS::Region')}:${aws_cdk_lib_1.Fn.ref('AWS::AccountId')}:stack/${props.stackName}*`,
                    ],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'cloudformation:CreateChangeSet',
                    ],
                    resources: [
                        `arn:aws:cloudformation:${aws_cdk_lib_1.Fn.ref('AWS::Region')}:aws:transform/Serverless-*`,
                    ],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'serverlessrepo:CreateCloudFormationTemplate',
                        'serverlessrepo:GetCloudFormationTemplate',
                    ],
                    resources: [
                        `arn:aws:serverlessrepo:${aws_cdk_lib_1.Fn.ref('AWS::Region')}:637378239786:applications/vws-aws-log-group-manager`,
                    ],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'ssm:GetParameters',
                    ],
                    resources: [`arn:aws:ssm:${aws_cdk_lib_1.Fn.ref('AWS::Region')}:${aws_cdk_lib_1.Fn.ref('AWS::AccountId')}:parameter/cdk-bootstrap/*`],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'iam:CreateRole',
                        'iam:GetRole',
                        'iam:DeleteRole',
                        'iam:AttachRolePolicy',
                        'iam:DetachRolePolicy',
                        'iam:PassRole',
                    ],
                    resources: [`arn:aws:iam::${aws_cdk_lib_1.Fn.ref('AWS::AccountId')}:role/${props.stackName}*`],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'iam:CreatePolicy',
                        'iam:GetPolicy',
                        'iam:ListPolicyVersions',
                        'iam:DeletePolicy',
                    ],
                    resources: [`arn:aws:iam::${aws_cdk_lib_1.Fn.ref('AWS::AccountId')}:policy/${props.stackName}*`],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'events:PutRule',
                        'events:PutTargets',
                        'events:RemoveTargets',
                        'events:DeleteRule',
                        'events:DescribeRule',
                    ],
                    resources: [`arn:aws:events:${aws_cdk_lib_1.Fn.ref('AWS::Region')}:${aws_cdk_lib_1.Fn.ref('AWS::AccountId')}:rule/${props.stackName}*`],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'kms:CreateKey',
                        'kms:TagResource',
                        'kms:PutKeyPolicy',
                        'kms:EnableKeyRotation',
                        'kms:DescribeKey',
                        'kms:CreateAlias',
                        'kms:DeleteAlias',
                    ],
                    resources: ['*'],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        'lambda:CreateFunction',
                        'lambda:CreateAlias',
                        'lambda:DeleteFunction',
                        'lambda:DeleteAlias',
                        'lambda:PutFunctionConcurrency',
                        'lambda:GetFunction',
                        'lambda:RemovePermission',
                        'lambda:ListVersionsByFunction',
                        'lambda:PublishVersion',
                        'lambda:AddPermission',
                    ],
                    resources: [
                        `arn:aws:lambda:${aws_cdk_lib_1.Fn.ref('AWS::Region')}:${aws_cdk_lib_1.Fn.ref('AWS::AccountId')}:function:${props.stackName}*`,
                    ],
                }),
                new aws_iam_1.PolicyStatement({
                    effect: aws_iam_1.Effect.ALLOW,
                    actions: [
                        's3:GetObject',
                    ],
                    resources: [
                        'arn:aws:s3:::awsserverlessrepo-changesets*/*',
                    ],
                }),
            ],
        };
        super(scope, id, policyProps);
    }
}
exports.LogGroupManagerDeploymentPolicy = LogGroupManagerDeploymentPolicy;
